<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.40">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>08 - Comunicação célula-célula – SC-COURSE</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting-549806ee2085284f45b00abea8c6df48.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap-859d446df91ed362524820d8142de709.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script src="../../site_libs/quarto-contrib/glightbox/glightbox.min.js"></script>
<link href="../../site_libs/quarto-contrib/glightbox/glightbox.min.css" rel="stylesheet">
<link href="../../site_libs/quarto-contrib/glightbox/lightbox.css" rel="stylesheet">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="../../styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../../index.html">
    <span class="navbar-title">SC-COURSE</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../index.html"> 
<span class="menu-text">Início</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../about.html"> 
<span class="menu-text">Sobre</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../projeto.html"> 
<span class="menu-text">Projeto</span></a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-semana-1" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Semana 1</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-semana-1">    
        <li>
    <a class="dropdown-item" href="../../material/semana1/01_data_collection.html">
 <span class="dropdown-text">Coleta de Dados</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../material/semana1/02_alignment.html">
 <span class="dropdown-text">Alinhamento</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-semana-2" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Semana 2</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-semana-2">    
        <li>
    <a class="dropdown-item" href="../../material/semana2/03_quality_control.html">
 <span class="dropdown-text">Controle de qualidade</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../material/semana2/04_clustering.html">
 <span class="dropdown-text">Clusterização</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-semana-3" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Semana 3</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-semana-3">    
        <li>
    <a class="dropdown-item" href="../../material/semana3/05_cell_annotation.html">
 <span class="dropdown-text">Anotação de células I</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../material/semana3/06_cell_annotation.html">
 <span class="dropdown-text">Anotação de células II</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-semana-4" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Semana 4</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-semana-4">    
        <li>
    <a class="dropdown-item" href="../../material/semana4/07_data_integration.html">
 <span class="dropdown-text">Integração de dados</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../material/semana4/08_cell_communication.html">
 <span class="dropdown-text">Comunicação célula-célula</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-semana-5" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Semana 5</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-semana-5">    
        <li>
    <a class="dropdown-item" href="../../material/semana5/09_cell_trajectory.html">
 <span class="dropdown-text">Análise de Trajetória</span></a>
  </li>  
    </ul>
  </li>
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#motivação" id="toc-motivação" class="nav-link active" data-scroll-target="#motivação">Motivação</a></li>
  <li><a href="#contextualização" id="toc-contextualização" class="nav-link" data-scroll-target="#contextualização">Contextualização</a></li>
  <li><a href="#ajustar-ambiente-e-carregar-pacotes" id="toc-ajustar-ambiente-e-carregar-pacotes" class="nav-link" data-scroll-target="#ajustar-ambiente-e-carregar-pacotes">Ajustar ambiente e carregar pacotes</a></li>
  <li><a href="#carregando-dados-do-projeto" id="toc-carregando-dados-do-projeto" class="nav-link" data-scroll-target="#carregando-dados-do-projeto">Carregando dados do projeto</a></li>
  <li><a href="#preparando-análise-de-comunicação" id="toc-preparando-análise-de-comunicação" class="nav-link" data-scroll-target="#preparando-análise-de-comunicação">Preparando análise de comunicação</a>
  <ul class="collapse">
  <li><a href="#liana-recursos" id="toc-liana-recursos" class="nav-link" data-scroll-target="#liana-recursos">LIANA: Recursos</a></li>
  <li><a href="#liana-métodos" id="toc-liana-métodos" class="nav-link" data-scroll-target="#liana-métodos">LIANA: Métodos</a></li>
  </ul></li>
  <li><a href="#calculando-interações-l-r" id="toc-calculando-interações-l-r" class="nav-link" data-scroll-target="#calculando-interações-l-r">Calculando interações L-R</a></li>
  <li><a href="#agregando-resultados" id="toc-agregando-resultados" class="nav-link" data-scroll-target="#agregando-resultados">Agregando resultados</a></li>
  <li><a href="#visualizar-conexões" id="toc-visualizar-conexões" class="nav-link" data-scroll-target="#visualizar-conexões">Visualizar conexões</a></li>
  <li><a href="#heatmap-com-interações-por-tipo-celular" id="toc-heatmap-com-interações-por-tipo-celular" class="nav-link" data-scroll-target="#heatmap-com-interações-por-tipo-celular">Heatmap com interações por tipo celular</a></li>
  <li><a href="#salvando-o-objeto-anotado" id="toc-salvando-o-objeto-anotado" class="nav-link" data-scroll-target="#salvando-o-objeto-anotado">Salvando o objeto anotado</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content page-columns page-full" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">08 - Comunicação célula-célula</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<section id="motivação" class="level2">
<h2 class="anchored" data-anchor-id="motivação">Motivação</h2>
<p>A comunicação célula-célula (CCC) é um processo fundamental na regulação de funções biológicas em tecidos complexos. Em contextos fisiológicos e patológicos, como no desenvolvimento embrionário, resposta imune e progressão tumoral, as células trocam sinais por meio de interações entre ligantes e receptores. O avanço das tecnologias de transcriptômica de célula única possibilitou a inferência sistemática desses eventos de comunicação, utilizando a expressão gênica como proxy para abundância proteica. Neste tutorial, exploramos o uso da ferramenta LIANA para realizar uma análise robusta e comparativa de interações célula-célula com base em múltiplos métodos e recursos anotados.</p>
</section>
<section id="contextualização" class="level2">
<h2 class="anchored" data-anchor-id="contextualização">Contextualização</h2>
<p>O pacote <strong>LIANA</strong> (Ligand-Receptor Analysis Framework) foi desenvolvido para facilitar essa tarefa ao integrar múltiplos algoritmos e recursos de interações ligante-receptor de forma padronizada. Com suporte nativo a objetos <code>Seurat</code> e <code>SingleCellExperiment</code>, LIANA permite a execução de métodos consagrados como CellPhoneDB, NATMI, Connectome, logFC, entre outros, utilizando bases como OmniPath, CellChatDB e Consensus. Além disso, oferece funções para agregação de resultados e geração de scores de consenso, promovendo maior robustez e confiança na inferência de eventos de comunicação celular.</p>
</section>
<section id="ajustar-ambiente-e-carregar-pacotes" class="level2">
<h2 class="anchored" data-anchor-id="ajustar-ambiente-e-carregar-pacotes">Ajustar ambiente e carregar pacotes</h2>
<p>Nesta seção, vamos configurar o ambiente de trabalho e carregar os pacotes necessários. Isso inclui verificar e instalar o pacote <code>liana</code>, além de carregar bibliotecas como <code>Seurat</code>, <code>tidyverse</code> e <code>ggplot2</code>, que serão utilizadas em todas as etapas subsequentes da análise.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Verifica pacotes</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">"liana"</span>)) {</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>  remotes<span class="sc">::</span><span class="fu">install_github</span>(<span class="st">'saezlab/liana'</span>)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Carregar pacotes</span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Seurat)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SeuratWrappers)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(liana)</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Definindo diretório padrão</span></span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>opts_knit<span class="sc">$</span><span class="fu">set</span>(</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a>  <span class="at">root.dir =</span> <span class="st">"/home/oandrefonseca/Disciplinas/PPGBM0117.2025.1"</span>,</span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a>  <span class="at">verbose =</span> <span class="cn">FALSE</span></span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Aumentar o limite de uso de memória</span></span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">future.globals.maxSize =</span> <span class="dv">10</span> <span class="sc">*</span> <span class="dv">1024</span><span class="sc">^</span><span class="dv">3</span>) <span class="co"># 10 GB, por exemplo</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="carregando-dados-do-projeto" class="level2 page-columns page-full">
<h2 class="anchored" data-anchor-id="carregando-dados-do-projeto">Carregando dados do projeto</h2>
<p>Agora que o ambiente está configurado, carregamos o objeto <code>Seurat</code> que contém os dados de expressão gênica já processados e anotados. O uso de um objeto pronto permite focar diretamente na inferência de comunicação célula-célula, utilizando as anotações de tipos celulares para identificar emissores e receptores em potenciais interações.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Carregando um objeto Seurat já processado (pré-clusterizado)</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>seurat_object <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data/seurat_annotated.RDS"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Verificar os clusters</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>  seurat_object, </span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">group.by =</span> <span class="st">"manual_annotation"</span></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell quarto-layout-panel page-columns page-full" data-layout-nrow="1" data-layout-align="center">
<div class="quarto-layout-row page-full">
<div class="cell-output-display quarto-layout-cell page-columns page-full" style="flex-basis: 100.0%;justify-content: center;">
<div id="fig-umap" class="quarto-float quarto-figure quarto-figure-center anchored page-columns page-full" data-fig-align="center" width="672">
<figure class="quarto-float quarto-float-fig figure page-columns page-full">
<div aria-describedby="fig-umap-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca" class="page-columns page-full">
<a href="08_cell_communication_files/figure-html/fig-umap-1.png" class="lightbox page-columns page-full" data-gallery="quarto-lightbox-gallery-1" title="Figure&nbsp;1: "><img src="08_cell_communication_files/figure-html/fig-umap-1.png" id="fig-umap" class="img-fluid quarto-figure quarto-figure-center anchored figure-img column-screen-inset-shaded" width="672"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig quarto-uncaptioned" id="fig-umap-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;1
</figcaption>
</figure>
</div>
</div>
</div>
</div>
</section>
<section id="preparando-análise-de-comunicação" class="level2">
<h2 class="anchored" data-anchor-id="preparando-análise-de-comunicação">Preparando análise de comunicação</h2>
<p>Com os dados carregados, damos início à etapa de preparação para a inferência de comunicação célula-célula. Aqui, exploramos os recursos e métodos disponíveis no pacote <code>liana</code>, que nos permitirão escolher diferentes bancos de interações e algoritmos para calcular os scores entre pares de tipos celulares.</p>
<section id="liana-recursos" class="level3">
<h3 class="anchored" data-anchor-id="liana-recursos">LIANA: Recursos</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">show_resources</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "Default"          "Consensus"        "Baccin2019"       "CellCall"        
 [5] "CellChatDB"       "Cellinker"        "CellPhoneDB"      "CellTalkDB"      
 [9] "connectomeDB2020" "EMBRACE"          "Guide2Pharma"     "HPMR"            
[13] "ICELLNET"         "iTALK"            "Kirouac2010"      "LRdb"            
[17] "Ramilowski2015"   "OmniPath"         "MouseConsensus"  </code></pre>
</div>
</div>
</section>
<section id="liana-métodos" class="level3">
<h3 class="anchored" data-anchor-id="liana-métodos">LIANA: Métodos</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">show_methods</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "connectome"      "logfc"           "natmi"           "sca"            
 [5] "cellphonedb"     "cytotalk"        "call_squidpy"    "call_cellchat"  
 [9] "call_connectome" "call_sca"        "call_italk"      "call_natmi"     </code></pre>
</div>
</div>
</section>
</section>
<section id="calculando-interações-l-r" class="level2">
<h2 class="anchored" data-anchor-id="calculando-interações-l-r">Calculando interações L-R</h2>
<p>Após observar que recursos e métodos podemos utilizar, vamos aplicar a função <code>liana_wrap()</code> para realizar a inferência das interações ligante-receptor entre os tipos celulares anotados no objeto Seurat. Esta etapa é central para o pipeline, pois gera a matriz de interações com scores específicos por método.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>liana_object <span class="ot">&lt;-</span> <span class="fu">liana_wrap</span>(</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>  seurat_object,</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">idents_col =</span> <span class="st">"manual_annotation"</span></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Expression from the `RNA` assay will be used</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Running LIANA with `manual_annotation` as labels!</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>`Idents` were converted to factor</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in exec(output, ...): 3031 genes and/or 0 cells were removed as they
had no counts!</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: `invoke()` is deprecated as of rlang 0.4.0.
Please use `exec()` or `inject()` instead.
This warning is displayed once every 8 hours.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>LIANA: LR summary stats calculated!</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Now Running: Natmi</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Now Running: Connectome</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Now Running: Logfc</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Now Running: Sca</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Now Running: Cellphonedb</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: `progress_estimated()` was deprecated in dplyr 1.0.0.
ℹ The deprecated feature was likely used in the liana package.
  Please report the issue at &lt;https://github.com/saezlab/liana/issues&gt;.</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>liana_object <span class="sc">%&gt;%</span> </span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">glimpse</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>List of 5
 $ natmi      : tibble [40,173 × 14] (S3: tbl_df/tbl/data.frame)
  ..$ source          : chr [1:40173] "CD4+ T cells / TCM-like" "CD4+ T cells / TCM-like" "CD4+ T cells / TCM-like" "CD4+ T cells / TCM-like" ...
  .. ..- attr(*, "levels")= chr [1:14] "B cells" "CD4+ T cells / TCM-like" "CD8+ Cytotoxic T cells" "Endothelial Cells" ...
  ..$ target          : chr [1:40173] "CD4+ T cells / TCM-like" "CD4+ T cells / TCM-like" "CD4+ T cells / TCM-like" "CD4+ T cells / TCM-like" ...
  .. ..- attr(*, "levels")= chr [1:14] "B cells" "CD4+ T cells / TCM-like" "CD8+ Cytotoxic T cells" "Endothelial Cells" ...
  ..$ ligand.complex  : chr [1:40173] "ADAM10" "ADAM10" "TNFSF14" "HLA-B" ...
  ..$ ligand          : chr [1:40173] "ADAM10" "ADAM10" "TNFSF14" "HLA-B" ...
  ..$ receptor.complex: chr [1:40173] "CD44" "NOTCH2" "TNFRSF14" "CD8A" ...
  ..$ receptor        : chr [1:40173] "CD44" "NOTCH2" "TNFRSF14" "CD8A" ...
  ..$ receptor.prop   : num [1:40173] 0.826 0.125 0.229 0.186 0.5 ...
  ..$ ligand.prop     : num [1:40173] 0.209 0.209 0.16 0.934 0.934 ...
  ..$ ligand.expr     : num [1:40173] 0.322 0.322 0.236 3.429 3.429 ...
  ..$ receptor.expr   : num [1:40173] 1.902 0.18 0.32 0.438 0.861 ...
  ..$ ligand.sum      : num [1:40173] 5.05 5.05 1.6 42.87 42.87 ...
  ..$ receptor.sum    : num [1:40173] 19.96 3.83 4.49 2.87 5.19 ...
  ..$ prod_weight     : num [1:40173] 0.6131 0.0579 0.0756 1.5003 2.9517 ...
  ..$ edge_specificity: num [1:40173] 0.00609 0.00299 0.01056 0.01218 0.01327 ...
 $ connectome : tibble [40,173 × 11] (S3: tbl_df/tbl/data.frame)
  ..$ source          : chr [1:40173] "CD4+ T cells / TCM-like" "CD4+ T cells / TCM-like" "CD4+ T cells / TCM-like" "CD4+ T cells / TCM-like" ...
  .. ..- attr(*, "levels")= chr [1:14] "B cells" "CD4+ T cells / TCM-like" "CD8+ Cytotoxic T cells" "Endothelial Cells" ...
  ..$ target          : chr [1:40173] "CD4+ T cells / TCM-like" "CD4+ T cells / TCM-like" "CD4+ T cells / TCM-like" "CD4+ T cells / TCM-like" ...
  .. ..- attr(*, "levels")= chr [1:14] "B cells" "CD4+ T cells / TCM-like" "CD8+ Cytotoxic T cells" "Endothelial Cells" ...
  ..$ ligand.complex  : chr [1:40173] "ADAM10" "ADAM10" "TNFSF14" "HLA-B" ...
  ..$ ligand          : chr [1:40173] "ADAM10" "ADAM10" "TNFSF14" "HLA-B" ...
  ..$ receptor.complex: chr [1:40173] "CD44" "NOTCH2" "TNFRSF14" "CD8A" ...
  ..$ receptor        : chr [1:40173] "CD44" "NOTCH2" "TNFRSF14" "CD8A" ...
  ..$ receptor.prop   : num [1:40173] 0.826 0.125 0.229 0.186 0.5 ...
  ..$ ligand.prop     : num [1:40173] 0.209 0.209 0.16 0.934 0.934 ...
  ..$ ligand.scaled   : num [1:40173] -0.0184 -0.0184 0.1631 -0.0692 -0.0692 ...
  ..$ receptor.scaled : num [1:40173] 0.2218 -0.1304 -0.0908 -0.0816 0.1269 ...
  ..$ weight_sc       : num [1:40173] 0.1017 -0.0744 0.0362 -0.0754 0.0289 ...
 $ logfc      : tibble [40,173 × 11] (S3: tbl_df/tbl/data.frame)
  ..$ source          : chr [1:40173] "CD4+ T cells / TCM-like" "CD4+ T cells / TCM-like" "CD4+ T cells / TCM-like" "CD4+ T cells / TCM-like" ...
  .. ..- attr(*, "levels")= chr [1:14] "B cells" "CD4+ T cells / TCM-like" "CD8+ Cytotoxic T cells" "Endothelial Cells" ...
  ..$ target          : chr [1:40173] "CD4+ T cells / TCM-like" "CD4+ T cells / TCM-like" "CD4+ T cells / TCM-like" "CD4+ T cells / TCM-like" ...
  .. ..- attr(*, "levels")= chr [1:14] "B cells" "CD4+ T cells / TCM-like" "CD8+ Cytotoxic T cells" "Endothelial Cells" ...
  ..$ ligand.complex  : chr [1:40173] "ADAM10" "ADAM10" "TNFSF14" "HLA-B" ...
  ..$ ligand          : chr [1:40173] "ADAM10" "ADAM10" "TNFSF14" "HLA-B" ...
  ..$ receptor.complex: chr [1:40173] "CD44" "NOTCH2" "TNFRSF14" "CD8A" ...
  ..$ receptor        : chr [1:40173] "CD44" "NOTCH2" "TNFRSF14" "CD8A" ...
  ..$ receptor.prop   : num [1:40173] 0.826 0.125 0.229 0.186 0.5 ...
  ..$ ligand.prop     : num [1:40173] 0.209 0.209 0.16 0.934 0.934 ...
  ..$ ligand.log2FC   : num [1:40173] 0.237 0.237 0.208 -0.218 -0.218 ...
  ..$ receptor.log2FC : num [1:40173] 0.5184 -0.0712 -0.0385 -0.3203 0.0845 ...
  ..$ logfc_comb      : num [1:40173] 0.3776 0.0828 0.0848 -0.2693 -0.0669 ...
 $ sca        : tibble [40,173 × 12] (S3: tbl_df/tbl/data.frame)
  ..$ source          : chr [1:40173] "CD4+ T cells / TCM-like" "CD4+ T cells / TCM-like" "CD4+ T cells / TCM-like" "CD4+ T cells / TCM-like" ...
  .. ..- attr(*, "levels")= chr [1:14] "B cells" "CD4+ T cells / TCM-like" "CD8+ Cytotoxic T cells" "Endothelial Cells" ...
  ..$ target          : chr [1:40173] "CD4+ T cells / TCM-like" "CD4+ T cells / TCM-like" "CD4+ T cells / TCM-like" "CD4+ T cells / TCM-like" ...
  .. ..- attr(*, "levels")= chr [1:14] "B cells" "CD4+ T cells / TCM-like" "CD8+ Cytotoxic T cells" "Endothelial Cells" ...
  ..$ ligand.complex  : chr [1:40173] "ADAM10" "ADAM10" "TNFSF14" "HLA-B" ...
  ..$ ligand          : chr [1:40173] "ADAM10" "ADAM10" "TNFSF14" "HLA-B" ...
  ..$ receptor.complex: chr [1:40173] "CD44" "NOTCH2" "TNFRSF14" "CD8A" ...
  ..$ receptor        : chr [1:40173] "CD44" "NOTCH2" "TNFRSF14" "CD8A" ...
  ..$ receptor.prop   : num [1:40173] 0.826 0.125 0.229 0.186 0.5 ...
  ..$ ligand.prop     : num [1:40173] 0.209 0.209 0.16 0.934 0.934 ...
  ..$ ligand.expr     : num [1:40173] 0.322 0.322 0.236 3.429 3.429 ...
  ..$ receptor.expr   : num [1:40173] 1.902 0.18 0.32 0.438 0.861 ...
  ..$ global_mean     : num [1:40173] 0.0741 0.0741 0.0741 0.0741 0.0741 ...
  ..$ LRscore         : num [1:40173] 0.914 0.764 0.788 0.943 0.959 ...
 $ cellphonedb: tibble [40,173 × 12] (S3: tbl_df/tbl/data.frame)
  ..$ source          : chr [1:40173] "CD4+ T cells / TCM-like" "CD4+ T cells / TCM-like" "CD4+ T cells / TCM-like" "CD4+ T cells / TCM-like" ...
  .. ..- attr(*, "levels")= chr [1:14] "B cells" "CD4+ T cells / TCM-like" "CD8+ Cytotoxic T cells" "Endothelial Cells" ...
  ..$ target          : chr [1:40173] "CD4+ T cells / TCM-like" "CD4+ T cells / TCM-like" "CD4+ T cells / TCM-like" "CD4+ T cells / TCM-like" ...
  .. ..- attr(*, "levels")= chr [1:14] "B cells" "CD4+ T cells / TCM-like" "CD8+ Cytotoxic T cells" "Endothelial Cells" ...
  ..$ ligand.complex  : chr [1:40173] "ADAM10" "ADAM10" "TNFSF14" "HLA-B" ...
  ..$ ligand          : chr [1:40173] "ADAM10" "ADAM10" "TNFSF14" "HLA-B" ...
  ..$ receptor.complex: chr [1:40173] "CD44" "NOTCH2" "TNFRSF14" "CD8A" ...
  ..$ receptor        : chr [1:40173] "CD44" "NOTCH2" "TNFRSF14" "CD8A" ...
  ..$ receptor.prop   : num [1:40173] 0.826 0.125 0.229 0.186 0.5 ...
  ..$ ligand.prop     : num [1:40173] 0.209 0.209 0.16 0.934 0.934 ...
  ..$ ligand.expr     : num [1:40173] 0.322 0.322 0.236 3.429 3.429 ...
  ..$ receptor.expr   : num [1:40173] 1.902 0.18 0.32 0.438 0.861 ...
  ..$ lr.mean         : num [1:40173] 1.112 0.251 0.278 1.933 2.145 ...
  ..$ pvalue          : num [1:40173] 0 1 0 1 0 0.017 1 1 1 1 ...</code></pre>
</div>
</div>
</section>
<section id="agregando-resultados" class="level2">
<h2 class="anchored" data-anchor-id="agregando-resultados">Agregando resultados</h2>
<p>Os resultados de múltiplos métodos podem variar. Para obter maior robustez, agregamos os resultados por meio de um rank de <strong>consenso</strong> utilizando <code>liana_aggregate()</code>. Essa abordagem permite priorizar interações que são consistentemente bem ranqueadas entre os diferentes algoritmos aplicados.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a>liana_object <span class="ot">&lt;-</span> liana_object <span class="sc">%&gt;%</span></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">liana_aggregate</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Now aggregating natmi</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Now aggregating connectome</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Now aggregating logfc</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Now aggregating sca</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Now aggregating cellphonedb</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Aggregating Ranks</code></pre>
</div>
</div>
</section>
<section id="visualizar-conexões" class="level2 page-columns page-full">
<h2 class="anchored" data-anchor-id="visualizar-conexões">Visualizar conexões</h2>
<p>Com os resultados agregados, passamos à etapa de visualização. Utilizamos a função <code>liana_dotplot()</code> para representar graficamente as interações priorizadas entre tipos celulares emissores e receptores. Esse tipo de plotagem facilita a interpretação das principais vias de sinalização inferidas no sistema biológico estudado.</p>
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a>liana_object <span class="sc">%&gt;%</span></span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">liana_dotplot</span>(</span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">source_groups =</span> <span class="fu">c</span>(<span class="st">"PRAME+ Tumor"</span>),</span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">target_groups =</span> <span class="fu">c</span>(<span class="st">"CD4+ T cells / TCM-like"</span>, <span class="st">"CD8+ Cytotoxic T cells"</span>, <span class="st">"B cells"</span>),</span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">ntop =</span> <span class="dv">20</span>) <span class="sc">+</span></span>
<span id="cb30-6"><a href="#cb30-6" aria-hidden="true" tabindex="-1"></a>  ggplot2<span class="sc">::</span><span class="fu">theme</span>(</span>
<span id="cb30-7"><a href="#cb30-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>, <span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>)</span>
<span id="cb30-8"><a href="#cb30-8" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell quarto-layout-panel page-columns page-full" data-layout-nrow="1" data-layout-align="center">
<div class="quarto-layout-row page-full">
<div class="cell-output-display quarto-layout-cell page-columns page-full" style="flex-basis: 100.0%;justify-content: center;">
<div id="fig-dotplot" class="quarto-float quarto-figure quarto-figure-center anchored page-columns page-full" data-fig-align="center" width="768">
<figure class="quarto-float quarto-float-fig figure page-columns page-full">
<div aria-describedby="fig-dotplot-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca" class="page-columns page-full">
<a href="08_cell_communication_files/figure-html/fig-dotplot-1.png" class="lightbox page-columns page-full" data-gallery="quarto-lightbox-gallery-2" title="Figure&nbsp;2: "><img src="08_cell_communication_files/figure-html/fig-dotplot-1.png" id="fig-dotplot" class="img-fluid quarto-figure quarto-figure-center anchored figure-img column-screen-inset-shaded" width="768"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig quarto-uncaptioned" id="fig-dotplot-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;2
</figcaption>
</figure>
</div>
</div>
</div>
</div>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>Cada ponto representa uma interação ligante-receptor priorizada. O eixo Y lista as interações (ligante → receptor) e o eixo X mostra os pares de tipos celulares. O tamanho do ponto indica a especificidade da interação, enquanto a intensidade da cor representa a magnitude do sinal (score de expressão). Interações mais relevantes tendem a ser grandes e escuras.</p>
</div>
</div>
</section>
<section id="heatmap-com-interações-por-tipo-celular" class="level2 page-columns page-full">
<h2 class="anchored" data-anchor-id="heatmap-com-interações-por-tipo-celular">Heatmap com interações por tipo celular</h2>
<p>Para uma visão global da frequência de interações entre tipos celulares, geramos um heatmap utilizando apenas interações altamente ranqueadas (<code>aggregate_rank &lt;= 0.01</code>). Essa visualização permite identificar padrões gerais de comunicação e potenciais hubs de sinalização no microambiente celular.</p>
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a>liana_trunc <span class="ot">&lt;-</span> liana_object <span class="sc">%&gt;%</span></span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(aggregate_rank <span class="sc">&lt;=</span> <span class="fl">0.01</span>)</span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Gerando visualização</span></span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a><span class="fu">heat_freq</span>(liana_trunc)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell quarto-layout-panel page-columns page-full" data-layout-nrow="1" data-layout-align="center">
<div class="quarto-layout-row page-full">
<div class="cell-output-display quarto-layout-cell page-columns page-full" style="flex-basis: 100.0%;justify-content: center;">
<div id="fig-heatmap" class="quarto-float quarto-figure quarto-figure-center anchored page-columns page-full" data-fig-align="center" width="768">
<figure class="quarto-float quarto-float-fig figure page-columns page-full">
<div aria-describedby="fig-heatmap-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca" class="page-columns page-full">
<a href="08_cell_communication_files/figure-html/fig-heatmap-1.png" class="lightbox page-columns page-full" data-gallery="quarto-lightbox-gallery-3" title="Figure&nbsp;3: "><img src="08_cell_communication_files/figure-html/fig-heatmap-1.png" id="fig-heatmap" class="img-fluid quarto-figure quarto-figure-center anchored figure-img column-screen-inset-shaded" width="768"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig quarto-uncaptioned" id="fig-heatmap-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3
</figcaption>
</figure>
</div>
</div>
</div>
</div>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>Neste gráfico, a intensidade da cor representa o número de interações inferidas entre os tipos celulares do eixo X (emissores) e do eixo Y (receptores). Tons mais escuros indicam maior número de interações inferidas entre os pares. Essa abordagem é útil para destacar populações celulares com papel central na rede de comunicação.</p>
</div>
</div>
</section>
<section id="salvando-o-objeto-anotado" class="level2">
<h2 class="anchored" data-anchor-id="salvando-o-objeto-anotado">Salvando o objeto anotado</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(seurat_object, <span class="at">file =</span> <span class="st">"data/seurat_annotated.RDS"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp("https:\/\/lint-edu\.github\.io\/PPGBM0117\.2025\.1");
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">
      &nbsp;
    </div>   
    <div class="nav-footer-center">
<p>Feito com ❤️ usando <a href="https://quarto.org">Quarto</a></p>
</div>
    <div class="nav-footer-right">
      &nbsp;
    </div>
  </div>
</footer>
<script>var lightboxQuarto = GLightbox({"closeEffect":"zoom","descPosition":"bottom","loop":false,"openEffect":"zoom","selector":".lightbox"});
(function() {
  let previousOnload = window.onload;
  window.onload = () => {
    if (previousOnload) {
      previousOnload();
    }
    lightboxQuarto.on('slide_before_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      const href = trigger.getAttribute('href');
      if (href !== null) {
        const imgEl = window.document.querySelector(`a[href="${href}"] img`);
        if (imgEl !== null) {
          const srcAttr = imgEl.getAttribute("src");
          if (srcAttr && srcAttr.startsWith("data:")) {
            slideConfig.href = srcAttr;
          }
        }
      } 
    });
  
    lightboxQuarto.on('slide_after_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(slideNode);
      }
    });
  
  };
  
})();
          </script>




</body></html>