<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.40">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>03 - Controle de qualidade – Treinamento em análise de célula única</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting-549806ee2085284f45b00abea8c6df48.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap-859d446df91ed362524820d8142de709.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script src="../../site_libs/quarto-contrib/glightbox/glightbox.min.js"></script>
<link href="../../site_libs/quarto-contrib/glightbox/glightbox.min.css" rel="stylesheet">
<link href="../../site_libs/quarto-contrib/glightbox/lightbox.css" rel="stylesheet">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<link href="../../site_libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet">
<script src="../../site_libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="../../site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="../../site_libs/datatables-binding-0.33/datatables.js"></script>
<script src="../../site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<link href="../../site_libs/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="../../site_libs/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="../../site_libs/dt-core-1.13.6/js/jquery.dataTables.min.js"></script>
<link href="../../site_libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet">
<script src="../../site_libs/crosstalk-1.2.1/js/crosstalk.min.js"></script>


<link rel="stylesheet" href="../../styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../../index.html">
    <span class="navbar-title">Treinamento em análise de célula única</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../index.html"> 
<span class="menu-text">Início</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../about.html"> 
<span class="menu-text">Sobre</span></a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-semana-1" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Semana 1</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-semana-1">    
        <li>
    <a class="dropdown-item" href="../../material/semana1/01_data_collection.html">
 <span class="dropdown-text">Coleta de Dados</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../material/semana1/02_alignment.html">
 <span class="dropdown-text">Alinhamento</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-semana-2" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Semana 2</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-semana-2">    
        <li>
    <a class="dropdown-item" href="../../material/semana2/03_quality_control.html">
 <span class="dropdown-text">Controle de qualidade</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../material/semana2/04_clustering.html">
 <span class="dropdown-text">Clusterização</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-semana-3" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Semana 3</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-semana-3">    
        <li>
    <a class="dropdown-item" href="../../material/semana3/05_cell_annotation.html">
 <span class="dropdown-text">Anotação de células I</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../material/semana3/06_cell_annotation.html">
 <span class="dropdown-text">Anotação de células II</span></a>
  </li>  
    </ul>
  </li>
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#motivação" id="toc-motivação" class="nav-link active" data-scroll-target="#motivação">Motivação</a></li>
  <li><a href="#contextualização" id="toc-contextualização" class="nav-link" data-scroll-target="#contextualização">Contextualização</a></li>
  <li><a href="#o-que-é-seurat" id="toc-o-que-é-seurat" class="nav-link" data-scroll-target="#o-que-é-seurat">O que é Seurat?</a></li>
  <li><a href="#estrutura-do-objeto-seurat" id="toc-estrutura-do-objeto-seurat" class="nav-link" data-scroll-target="#estrutura-do-objeto-seurat">Estrutura do Objeto <code>Seurat</code></a></li>
  <li><a href="#descrevendo-o-metadata" id="toc-descrevendo-o-metadata" class="nav-link" data-scroll-target="#descrevendo-o-metadata">Descrevendo o metadata</a></li>
  <li><a href="#avaliação-da-qualidade-das-células" id="toc-avaliação-da-qualidade-das-células" class="nav-link" data-scroll-target="#avaliação-da-qualidade-das-células">Avaliação da qualidade das células</a>
  <ul class="collapse">
  <li><a href="#calculando-percentagem-relacionada-a-genes-de-interesse" id="toc-calculando-percentagem-relacionada-a-genes-de-interesse" class="nav-link" data-scroll-target="#calculando-percentagem-relacionada-a-genes-de-interesse">Calculando percentagem relacionada a genes de interesse</a></li>
  <li><a href="#salvando-estatísticas-pré-filtragem" id="toc-salvando-estatísticas-pré-filtragem" class="nav-link" data-scroll-target="#salvando-estatísticas-pré-filtragem">Salvando estatísticas pré-filtragem</a></li>
  <li><a href="#visualizando-métricas-de-interesse" id="toc-visualizando-métricas-de-interesse" class="nav-link" data-scroll-target="#visualizando-métricas-de-interesse">Visualizando métricas de interesse</a></li>
  <li><a href="#correlacionando-métricas" id="toc-correlacionando-métricas" class="nav-link" data-scroll-target="#correlacionando-métricas">Correlacionando métricas</a></li>
  <li><a href="#identificando-limiares-de-corte" id="toc-identificando-limiares-de-corte" class="nav-link" data-scroll-target="#identificando-limiares-de-corte">Identificando limiares de corte</a></li>
  </ul></li>
  <li><a href="#identificando-dupletos" id="toc-identificando-dupletos" class="nav-link" data-scroll-target="#identificando-dupletos">Identificando dupletos</a>
  <ul class="collapse">
  <li><a href="#por-que-isso-é-um-problema" id="toc-por-que-isso-é-um-problema" class="nav-link" data-scroll-target="#por-que-isso-é-um-problema">Por que isso é um problema?</a></li>
  </ul></li>
  <li><a href="#filtrando-células-de-baixa-qualidade" id="toc-filtrando-células-de-baixa-qualidade" class="nav-link" data-scroll-target="#filtrando-células-de-baixa-qualidade">Filtrando células de baixa qualidade</a>
  <ul class="collapse">
  <li><a href="#salvando-estatística-pós-filtragem" id="toc-salvando-estatística-pós-filtragem" class="nav-link" data-scroll-target="#salvando-estatística-pós-filtragem">Salvando estatística pós-filtragem</a></li>
  <li><a href="#combinando-as-estatísticas-pré-e-pós-filtragem" id="toc-combinando-as-estatísticas-pré-e-pós-filtragem" class="nav-link" data-scroll-target="#combinando-as-estatísticas-pré-e-pós-filtragem">Combinando as estatísticas pré e pós-filtragem</a></li>
  </ul></li>
  <li><a href="#comparando-antes-e-depois" id="toc-comparando-antes-e-depois" class="nav-link" data-scroll-target="#comparando-antes-e-depois">Comparando antes e depois</a></li>
  <li><a href="#normalização-protocolo-padrão" id="toc-normalização-protocolo-padrão" class="nav-link" data-scroll-target="#normalização-protocolo-padrão">Normalização — Protocolo padrão</a>
  <ul class="collapse">
  <li><a href="#encontrando-genes-com-alta-variabilidade" id="toc-encontrando-genes-com-alta-variabilidade" class="nav-link" data-scroll-target="#encontrando-genes-com-alta-variabilidade">Encontrando genes com alta variabilidade</a></li>
  <li><a href="#ajustando-escala-dos-dados" id="toc-ajustando-escala-dos-dados" class="nav-link" data-scroll-target="#ajustando-escala-dos-dados">Ajustando escala dos dados</a></li>
  </ul></li>
  <li><a href="#normalização---sctransform" id="toc-normalização---sctransform" class="nav-link" data-scroll-target="#normalização---sctransform">Normalização - SCTransform</a></li>
  <li><a href="#redução-de-dimensionalidade" id="toc-redução-de-dimensionalidade" class="nav-link" data-scroll-target="#redução-de-dimensionalidade">Redução de dimensionalidade</a></li>
  <li><a href="#projeção-com-umap" id="toc-projeção-com-umap" class="nav-link" data-scroll-target="#projeção-com-umap">Projeção com UMAP</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content page-columns page-full" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">03 - Controle de qualidade</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<section id="motivação" class="level2">
<h2 class="anchored" data-anchor-id="motivação">Motivação</h2>
<p>Antes de realizar qualquer análise exploratória ou inferência estatística com dados de RNA-seq de célula única (scRNA-seq), é essencial garantir que os dados sejam de <strong>alta qualidade</strong>. Diferentemente do RNA-seq convencional (bulk), dados de scRNA-seq são altamente suscetíveis a <strong>ruído técnico</strong> e <strong>artefatos experimentais</strong>, como células mortas, dupletos e contaminação por RNA ambiente.</p>
</section>
<section id="contextualização" class="level2">
<h2 class="anchored" data-anchor-id="contextualização">Contextualização</h2>
<p>Dados de RNA-seq de célula única (scRNA-seq) são poderosos para investigar a heterogeneidade celular, mas exigem um rigoroso controle de qualidade devido à presença de ruídos técnicos, células danificadas, dupletos e contaminação por RNA ambiente.</p>
<p>Neste tutorial, utilizamos um conjunto de ferramentas complementares:</p>
<ul>
<li><strong><code>Seurat</code></strong> para pré-processamento, cálculo de métricas de qualidade e filtragem de células;</li>
<li><strong><code>SoupX</code></strong> para identificar e corrigir contaminação por RNA ambiente (ambient RNA);</li>
<li><strong><code>scDblFinder</code></strong> para detectar e remover dupletos (<em>doublets</em>).</li>
</ul>
<p>A aplicação combinada desses pacotes permite construir um conjunto de células de alta qualidade, garantindo que análises posteriores — como clusterização e anotação de tipos celulares — sejam mais robustas e confiáveis.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Seurat)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scDblFinder)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(here)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Definindo diretorio padrao</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>opts_knit<span class="sc">$</span><span class="fu">set</span>(</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">root.dir =</span> <span class="st">"/home/oandrefonseca/Disciplinas/PPGBM0117.2025.1"</span>,</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">verbose =</span> <span class="cn">FALSE</span></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="o-que-é-seurat" class="level2">
<h2 class="anchored" data-anchor-id="o-que-é-seurat">O que é Seurat?</h2>
<p>Seurat é um pacote R desenvolvido para controle de qualidade, análise e exploração de dados de RNA-sequenciamento de células individuais. Seurat visa permitir que os usuários identifiquem e interpretem fontes de heterogeneidade a partir de medições transcriptômicas de células individuais e integrem diversos tipos de dados de células individuais. Para mais detalhes acesse <a href="https://satijalab.org/seurat/">aqui</a>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>breast_data <span class="ot">&lt;-</span> <span class="fu">Read10X</span>(</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">data.dir =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data/filtered_feature_bc_matrix/"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>seurat_object <span class="ot">&lt;-</span> <span class="fu">CreateSeuratObject</span>(</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">counts =</span> breast_data, <span class="at">project =</span> <span class="st">"Breast_7k"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="estrutura-do-objeto-seurat" class="level2">
<h2 class="anchored" data-anchor-id="estrutura-do-objeto-seurat">Estrutura do Objeto <code>Seurat</code></h2>
<p>Para visualizar a estrutura do objeto recém-construídos, podemos utilizar o comando:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Exibe de forma compacta a estrutura interna de um objeto R</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(seurat_object)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<table class="caption-top table">
<colgroup>
<col style="width: 38%">
<col style="width: 36%">
<col style="width: 25%">
</colgroup>
<thead>
<tr class="header">
<th>Slot</th>
<th>Tipo de Objeto</th>
<th>Descrição</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>assays</code></td>
<td>Lista de <code>Assay</code></td>
<td>Armazena dados de expressão para diferentes modalidades (e.g., RNA, ATAC). Cada <code>Assay</code> contém camadas como <code>counts</code>, <code>data</code> e <code>scale.data</code>.</td>
</tr>
<tr class="even">
<td><code>meta.data</code></td>
<td><code>data.frame</code></td>
<td>Metadados por célula, como <code>nFeature_RNA</code>, <code>nCount_RNA</code>, <code>percent.mt</code>, entre outros.</td>
</tr>
<tr class="odd">
<td><code>active.assay</code></td>
<td><code>character</code></td>
<td>Nome do <code>Assay</code> ativo, utilizado por funções padrão como <code>NormalizeData()</code>.</td>
</tr>
<tr class="even">
<td><code>active.ident</code></td>
<td><code>factor</code></td>
<td>Identidade celular ativa, geralmente usada para clusterização e visualizações.</td>
</tr>
<tr class="odd">
<td><code>reductions</code></td>
<td>Lista de <code>DimReduc</code></td>
<td>Armazena resultados de reduções dimensionais, como PCA, UMAP e t-SNE.</td>
</tr>
<tr class="even">
<td><code>graphs</code></td>
<td>Lista de <code>Graph</code></td>
<td>Contém grafos de vizinhança utilizados em algoritmos de clusterização, como <code>FindNeighbors()</code>.</td>
</tr>
<tr class="odd">
<td><code>neighbors</code></td>
<td>Lista</td>
<td>Armazena informações sobre vizinhos mais próximos calculados, úteis para métodos como UMAP.</td>
</tr>
<tr class="even">
<td><code>images</code></td>
<td>Lista de <code>SpatialImage</code></td>
<td>Armazena dados de imagens espaciais, como os provenientes de experimentos Visium.</td>
</tr>
<tr class="odd">
<td><code>project.name</code></td>
<td><code>character</code></td>
<td>Nome do projeto associado ao objeto <code>Seurat</code>.</td>
</tr>
<tr class="even">
<td><code>misc</code></td>
<td>Lista</td>
<td>Informações diversas e adicionais que não se encaixam nos outros slots.</td>
</tr>
<tr class="odd">
<td><code>version</code></td>
<td><code>package_version</code></td>
<td>Versão do pacote <code>Seurat</code> utilizada para criar o objeto.</td>
</tr>
<tr class="even">
<td><code>commands</code></td>
<td>Lista de <code>SeuratCommand</code></td>
<td>Histórico de comandos aplicados ao objeto, incluindo parâmetros e timestamps.</td>
</tr>
<tr class="odd">
<td><code>tools</code></td>
<td>Lista</td>
<td>Resultados de análises adicionais geradas por outras ferramentas, preenchido por desenvolvedores usando <code>Tool&lt;-</code>.</td>
</tr>
</tbody>
</table>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-1-contents" aria-controls="callout-1" aria-expanded="true" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-1" class="callout-1-contents callout-collapse collapse show">
<div class="callout-body-container callout-body">
<p>A estrutura do objeto <code>Seurat</code> pode <strong>sofrer pequenas alterações entre versões do pacote</strong>, especialmente em relação à organização interna dos slots, nomenclatura e slots adicionais utilizados por novos métodos.</p>
<p>Este tutorial foi desenvolvido com a versão <strong>Seurat 5.2.1</strong>, na qual os principais componentes do objeto — como <code>assays</code>, <code>meta.data</code>, <code>reductions</code>, <code>graphs</code> e <code>commands</code> — mantêm compatibilidade com versões anteriores, mas apresentam melhorias de desempenho e integração com múltiplas modalidades.</p>
</div>
</div>
</div>
</section>
<section id="descrevendo-o-metadata" class="level2">
<h2 class="anchored" data-anchor-id="descrevendo-o-metadata">Descrevendo o metadata</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">View</span>(seurat_object<span class="sc">@</span>meta.data)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-2-contents" aria-controls="callout-2" aria-expanded="true" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-2" class="callout-2-contents callout-collapse collapse show">
<div class="callout-body-container callout-body">
<ul>
<li><strong>Número total de moleculas unicas (<code>nCount_RNA</code>)</strong><br>
</li>
<li><strong>Número de genes detectados (<code>nFeature_RNA</code>)</strong><br>
</li>
</ul>
</div>
</div>
</div>
</section>
<section id="avaliação-da-qualidade-das-células" class="level2 page-columns page-full">
<h2 class="anchored" data-anchor-id="avaliação-da-qualidade-das-células">Avaliação da qualidade das células</h2>
<section id="calculando-percentagem-relacionada-a-genes-de-interesse" class="level3">
<h3 class="anchored" data-anchor-id="calculando-percentagem-relacionada-a-genes-de-interesse">Calculando percentagem relacionada a genes de interesse</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Genes mitocondriais (nomes iniciam com "MT-")</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>seurat_object[[<span class="st">"percent.mt"</span>]] <span class="ot">&lt;-</span> <span class="fu">PercentageFeatureSet</span>(</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>  seurat_object, <span class="at">pattern =</span> <span class="st">"^MT-"</span>)</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Genes ribossômicos (nomes iniciam com "RPS" ou "RPL")</span></span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>seurat_object[[<span class="st">"percent.ribo"</span>]] <span class="ot">&lt;-</span> <span class="fu">PercentageFeatureSet</span>(</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>  seurat_object, <span class="at">pattern =</span> <span class="st">"^RP[SL]"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Genes de hemoglobina - remover HBP explicitamente</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>hb_genes <span class="ot">&lt;-</span> <span class="fu">grep</span>(<span class="st">"^HB"</span>, <span class="fu">rownames</span>(seurat_object), <span class="at">value =</span> <span class="cn">TRUE</span>)</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>hb_genes <span class="ot">&lt;-</span> hb_genes[<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">"^HBP"</span>, hb_genes)]</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>seurat_object[[<span class="st">"percent.hb"</span>]] <span class="ot">&lt;-</span> <span class="fu">PercentageFeatureSet</span>(</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>  seurat_object, <span class="at">features =</span> hb_genes)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-3-contents" aria-controls="callout-3" aria-expanded="true" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-3" class="callout-3-contents callout-collapse collapse show">
<div class="callout-body-container callout-body">
<p>Essas porcentagens ajudam a:</p>
<ul>
<li><strong>Filtrar células de baixa qualidade</strong> (ex: alto <code>percent.mt</code>);</li>
<li><strong>Identificar artefatos ou contaminações</strong> (ex: glóbulos vermelhos via <code>percent.hb</code>);</li>
<li><strong>Detectar viés técnico</strong> ou diferenças biológicas significativas (ex: via <code>percent.ribo</code>).</li>
</ul>
<p>Essas métricas são frequentemente visualizadas via <code>VlnPlot()</code> ou correlacionadas com <code>nCount_RNA</code> para detectar padrões suspeitos.</p>
</div>
</div>
</div>
</section>
<section id="salvando-estatísticas-pré-filtragem" class="level3">
<h3 class="anchored" data-anchor-id="salvando-estatísticas-pré-filtragem">Salvando estatísticas pré-filtragem</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>pre_qc_stats <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">number_of_cells =</span> <span class="fu">ncol</span>(seurat_object),</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">mean_counts_per_cell =</span> <span class="fu">mean</span>(seurat_object<span class="sc">$</span>nCount_RNA),</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">median_counts_per_cell =</span> <span class="fu">median</span>(seurat_object<span class="sc">$</span>nCount_RNA),</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">mean_features_per_cell =</span> <span class="fu">mean</span>(seurat_object<span class="sc">$</span>nFeature_RNA),</span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">median_features_per_cell =</span> <span class="fu">median</span>(seurat_object<span class="sc">$</span>nFeature_RNA)</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="visualizando-métricas-de-interesse" class="level3 page-columns page-full">
<h3 class="anchored" data-anchor-id="visualizando-métricas-de-interesse">Visualizando métricas de interesse</h3>
<div class="cell page-columns page-full" data-layout-align="center">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">VlnPlot</span>(seurat_object, </span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>        <span class="at">features =</span> <span class="fu">c</span>(<span class="st">"nCount_RNA"</span>, <span class="st">"nFeature_RNA"</span>), <span class="at">ncol =</span> <span class="dv">3</span>)</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">VlnPlot</span>(seurat_object, </span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>        <span class="at">features =</span> <span class="fu">c</span>(<span class="st">"percent.mt"</span>, <span class="st">"percent.ribo"</span>, <span class="st">"percent.hb"</span>), <span class="at">ncol =</span> <span class="dv">3</span>)</span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>p1</span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a>p2</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display page-columns page-full">
<div id="fig-quality-metrics-1" class="quarto-float quarto-figure quarto-figure-center anchored page-columns page-full" data-fig-align="center" width="672">
<figure class="quarto-float quarto-float-fig figure page-columns page-full">
<div aria-describedby="fig-quality-metrics-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca" class="page-columns page-full">
<a href="03_quality_control_files/figure-html/fig-quality-metrics-1.png" class="lightbox page-columns page-full" data-gallery="quarto-lightbox-gallery-1" title="Figure&nbsp;1: "><img src="03_quality_control_files/figure-html/fig-quality-metrics-1.png" id="fig-quality-metrics-1" class="img-fluid quarto-figure quarto-figure-center anchored figure-img column-screen-inset-shaded" width="672"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig quarto-uncaptioned" id="fig-quality-metrics-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;1
</figcaption>
</figure>
</div>
</div>
<div class="cell-output-display page-columns page-full">
<div id="fig-quality-metrics-2" class="quarto-float quarto-figure quarto-figure-center anchored page-columns page-full" data-fig-align="center" width="672">
<figure class="quarto-float quarto-float-fig figure page-columns page-full">
<div aria-describedby="fig-quality-metrics-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca" class="page-columns page-full">
<a href="03_quality_control_files/figure-html/fig-quality-metrics-2.png" class="lightbox page-columns page-full" data-gallery="quarto-lightbox-gallery-2" title="Figure&nbsp;2: "><img src="03_quality_control_files/figure-html/fig-quality-metrics-2.png" id="fig-quality-metrics-2" class="img-fluid quarto-figure quarto-figure-center anchored figure-img column-screen-inset-shaded" width="672"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig quarto-uncaptioned" id="fig-quality-metrics-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;2
</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="correlacionando-métricas" class="level3 page-columns page-full">
<h3 class="anchored" data-anchor-id="correlacionando-métricas">Correlacionando métricas</h3>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>plot1 <span class="ot">&lt;-</span> <span class="fu">FeatureScatter</span>(seurat_object, <span class="at">feature1 =</span> <span class="st">"nCount_RNA"</span>, <span class="at">feature2 =</span> <span class="st">"percent.mt"</span>)</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>plot2 <span class="ot">&lt;-</span> <span class="fu">FeatureScatter</span>(seurat_object, <span class="at">feature1 =</span> <span class="st">"nCount_RNA"</span>, <span class="at">feature2 =</span> <span class="st">"nFeature_RNA"</span>)</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>plot1</span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>plot2</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell column-screen-inset-shaded quarto-layout-panel" data-layout-nrow="1">
<div class="quarto-layout-row">
<div class="cell-output-display quarto-layout-cell" style="flex-basis: 50.0%;justify-content: flex-start;">
<div id="fig-quality-correlation-1" class="quarto-float quarto-figure quarto-figure-center anchored" width="672">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-quality-correlation-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="03_quality_control_files/figure-html/fig-quality-correlation-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-3" title="Figure&nbsp;3: "><img src="03_quality_control_files/figure-html/fig-quality-correlation-1.png" id="fig-quality-correlation-1" class="img-fluid figure-img" width="672"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig quarto-uncaptioned" id="fig-quality-correlation-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3
</figcaption>
</figure>
</div>
</div>
<div class="cell-output-display quarto-layout-cell" style="flex-basis: 50.0%;justify-content: flex-start;">
<div id="fig-quality-correlation-2" class="quarto-float quarto-figure quarto-figure-center anchored" width="672">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-quality-correlation-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="03_quality_control_files/figure-html/fig-quality-correlation-2.png" class="lightbox" data-gallery="quarto-lightbox-gallery-4" title="Figure&nbsp;4: "><img src="03_quality_control_files/figure-html/fig-quality-correlation-2.png" id="fig-quality-correlation-2" class="img-fluid figure-img" width="672"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig quarto-uncaptioned" id="fig-quality-correlation-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;4
</figcaption>
</figure>
</div>
</div>
</div>
</div>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-4-contents" aria-controls="callout-4" aria-expanded="true" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-4" class="callout-4-contents callout-collapse collapse show">
<div class="callout-body-container callout-body">
<ol type="1">
<li><p><strong><code>nCount_RNA</code> vs.&nbsp;<code>nFeature_RNA</code></strong><br>
Uma <strong>correlação positiva forte</strong> é esperada: quanto maior o número total de transcritos, maior tende a ser o número de genes detectados.<br>
→ Uma correlação <strong>anormalmente fraca</strong> pode indicar problemas técnicos ou presença de dupletos.</p></li>
<li><p><strong><code>nCount_RNA</code> vs.&nbsp;<code>percent.mt</code></strong><br>
Geralmente apresentam uma <strong>correlação negativa</strong>. Células com um número total maior de moléculas de RNA (alto nCount_RNA) são tipicamente associadas a uma porcentagem menor de leituras mitocondriais (menor percent.mt), indicando melhor qualidade celular e menos contaminação. → Uma <strong>alta percentagem mitocondrial com poucos counts</strong> pode ser um sinal de apoptose ou lise celular.</p></li>
</ol>
</div>
</div>
</div>
</section>
<section id="identificando-limiares-de-corte" class="level3">
<h3 class="anchored" data-anchor-id="identificando-limiares-de-corte">Identificando limiares de corte</h3>
<p>Esta função permite identificar <strong>células com valores extremos</strong> para uma métrica contínua, como <code>nFeature_RNA</code> ou <code>percent.mt</code>, utilizando um critério estatístico baseado em:</p>
<blockquote class="blockquote">
<p><strong>mediana ± (n × MAD)</strong></p>
</blockquote>
<p>Essa abordagem é <strong>mais robusta que o desvio padrão</strong>, pois é menos sensível a valores extremos.<br>
O argumento <code>nmads</code> controla a sensibilidade da filtragem (3 é o valor comumente utilizado).</p>
<p>Esta função é útil para <strong>detectar e remover artefatos experimentais</strong> ou células anômalas de forma personalizada.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>is_outlier <span class="ot">&lt;-</span> <span class="cf">function</span>(seurat_object, metric, <span class="at">nmads =</span> <span class="dv">3</span>, <span class="at">verbose =</span> <span class="cn">TRUE</span>) {</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extrai os valores da métrica desejada</span></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>  values <span class="ot">&lt;-</span> seurat_object<span class="sc">@</span>meta.data[[metric]]</span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Calcula mediana e desvio absoluto da mediana (MAD)</span></span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>  med <span class="ot">&lt;-</span> <span class="fu">median</span>(values, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>  mad_val <span class="ot">&lt;-</span> <span class="fu">mad</span>(values, <span class="at">center =</span> med, <span class="at">constant =</span> <span class="dv">1</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Define os thresholds inferior e superior</span></span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a>  lower <span class="ot">&lt;-</span> med <span class="sc">-</span> nmads <span class="sc">*</span> mad_val</span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a>  upper <span class="ot">&lt;-</span> med <span class="sc">+</span> nmads <span class="sc">*</span> mad_val</span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-13"><a href="#cb11-13" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Imprime os limites, se verbose = TRUE</span></span>
<span id="cb11-14"><a href="#cb11-14" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (verbose) {</span>
<span id="cb11-15"><a href="#cb11-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">message</span>(glue<span class="sc">::</span><span class="fu">glue</span>(<span class="st">"Outliers para '{metric}' definidos como valores fora do intervalo:"</span>))</span>
<span id="cb11-16"><a href="#cb11-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">message</span>(glue<span class="sc">::</span><span class="fu">glue</span>(<span class="st">"  Inferior: {round(lower, 2)}"</span>))</span>
<span id="cb11-17"><a href="#cb11-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">message</span>(glue<span class="sc">::</span><span class="fu">glue</span>(<span class="st">"  Superior: {round(upper, 2)}"</span>))</span>
<span id="cb11-18"><a href="#cb11-18" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb11-19"><a href="#cb11-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-20"><a href="#cb11-20" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Identifica outliers</span></span>
<span id="cb11-21"><a href="#cb11-21" aria-hidden="true" tabindex="-1"></a>  outlier <span class="ot">&lt;-</span> (values <span class="sc">&lt;</span> lower) <span class="sc">|</span> (values <span class="sc">&gt;</span> upper)</span>
<span id="cb11-22"><a href="#cb11-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(outlier)</span>
<span id="cb11-23"><a href="#cb11-23" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>seurat_object<span class="sc">$</span>mt_outlier <span class="ot">&lt;-</span> <span class="fu">is_outlier</span>(</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>  seurat_object, <span class="at">metric =</span> <span class="st">"percent.mt"</span>, <span class="at">nmads =</span> <span class="dv">3</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Outliers para 'percent.mt' definidos como valores fora do intervalo:</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>  Inferior: -15.07</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>  Superior: 41.41</code></pre>
</div>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(seurat_object<span class="sc">$</span>mt_outlier)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
FALSE  TRUE 
 3719  1961 </code></pre>
</div>
</div>
</section>
</section>
<section id="identificando-dupletos" class="level2">
<h2 class="anchored" data-anchor-id="identificando-dupletos">Identificando dupletos</h2>
<p><strong>Dupletos (doublets)</strong> ocorrem quando duas células distintas são capturadas dentro da mesma gotícula durante a etapa de microencapsulamento (como no 10x Genomics). Como resultado, seus perfis transcriptômicos são combinados e registrados como uma única célula.</p>
<section id="por-que-isso-é-um-problema" class="level3">
<h3 class="anchored" data-anchor-id="por-que-isso-é-um-problema">Por que isso é um problema?</h3>
<ul>
<li>Dupletos podem <strong>formar clusters artificiais</strong> ou <strong>preencher espaços entre populações reais</strong>, confundindo a interpretação biológica.</li>
<li>Em análises downstream, como <strong>clusterização</strong>, <strong>diferenciação de tipos celulares</strong> e <strong>descoberta de trajetórias</strong>, os dupletos introduzem <strong>ruído</strong>.</li>
</ul>
<p>Ferramentas como <strong><code>scDblFinder</code></strong>, <strong><code>Scrublet</code></strong> e <strong><code>DoubletFinder</code></strong> utilizam estratégias estatísticas e simulações para identificar prováveis dupletos com base na similaridade de perfis de expressão e distribuição de vizinhos.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Converta o objeto Seurat para SingleCellExperiment</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>sce <span class="ot">&lt;-</span> <span class="fu">as.SingleCellExperiment</span>(seurat_object)</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Executa a identificação de dupletos</span></span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>sce <span class="ot">&lt;-</span> <span class="fu">scDblFinder</span>(sce)</span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Adiciona o resultado de volta ao objeto Seurat</span></span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a>seurat_object<span class="sc">$</span>scDblFinder.class <span class="ot">&lt;-</span> sce<span class="sc">$</span>scDblFinder.class</span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a>seurat_object<span class="sc">$</span>doublet_score <span class="ot">&lt;-</span> sce<span class="sc">$</span>scDblFinder.score</span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualiza a proporção de dupletos identificados</span></span>
<span id="cb18-12"><a href="#cb18-12" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(seurat_object<span class="sc">$</span>scDblFinder.class)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
singlet doublet 
   5203     477 </code></pre>
</div>
</div>
</section>
</section>
<section id="filtrando-células-de-baixa-qualidade" class="level2">
<h2 class="anchored" data-anchor-id="filtrando-células-de-baixa-qualidade">Filtrando células de baixa qualidade</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Criando copia do objeto pre-filtragem</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>pre_seurat_object <span class="ot">&lt;-</span> seurat_object</span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Filtro com thresholds definidos manualmente</span></span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a>seurat_object <span class="ot">&lt;-</span> <span class="fu">subset</span>(</span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a>  seurat_object,</span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">subset =</span> nFeature_RNA <span class="sc">&gt;</span> <span class="dv">200</span> <span class="sc">&amp;</span></span>
<span id="cb20-8"><a href="#cb20-8" aria-hidden="true" tabindex="-1"></a>           nFeature_RNA <span class="sc">&lt;</span> <span class="dv">6000</span> <span class="sc">&amp;</span></span>
<span id="cb20-9"><a href="#cb20-9" aria-hidden="true" tabindex="-1"></a>           percent.mt <span class="sc">&lt;</span> <span class="dv">10</span> <span class="sc">&amp;</span></span>
<span id="cb20-10"><a href="#cb20-10" aria-hidden="true" tabindex="-1"></a>           scDblFinder.class <span class="sc">==</span> <span class="st">"singlet"</span></span>
<span id="cb20-11"><a href="#cb20-11" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="salvando-estatística-pós-filtragem" class="level3">
<h3 class="anchored" data-anchor-id="salvando-estatística-pós-filtragem">Salvando estatística pós-filtragem</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Estatísticas após o QC</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>post_qc_stats <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">number_of_cells =</span> <span class="fu">ncol</span>(seurat_object),</span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">mean_counts_per_cell =</span> <span class="fu">mean</span>(seurat_object<span class="sc">$</span>nCount_RNA),</span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">median_counts_per_cell =</span> <span class="fu">median</span>(seurat_object<span class="sc">$</span>nCount_RNA),</span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">mean_features_per_cell =</span> <span class="fu">mean</span>(seurat_object<span class="sc">$</span>nFeature_RNA),</span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">median_features_per_cell =</span> <span class="fu">median</span>(seurat_object<span class="sc">$</span>nFeature_RNA)</span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="combinando-as-estatísticas-pré-e-pós-filtragem" class="level3">
<h3 class="anchored" data-anchor-id="combinando-as-estatísticas-pré-e-pós-filtragem">Combinando as estatísticas pré e pós-filtragem</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Unindo as tabelas</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>qc_summary <span class="ot">&lt;-</span> <span class="fu">rbind</span>(</span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cbind</span>(<span class="at">Etapa =</span> <span class="st">"Antes do QC"</span>, pre_qc_stats),</span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cbind</span>(<span class="at">Etapa =</span> <span class="st">"Depois do QC"</span>, post_qc_stats)</span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="comparando-antes-e-depois" class="level2 page-columns page-full">
<h2 class="anchored" data-anchor-id="comparando-antes-e-depois">Comparando antes e depois</h2>
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a>DT<span class="sc">::</span><span class="fu">datatable</span>(qc_summary)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell column-screen-inset-shaded quarto-layout-panel" data-layout-nrow="1">
<div class="quarto-layout-row">
<div class="quarto-layout-cell" style="flex-basis: 100.0%;justify-content: flex-start;">
<div class="datatables html-widget html-fill-item" id="htmlwidget-f843a680a1fcd62a0dd1" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-f843a680a1fcd62a0dd1">{"x":{"filter":"none","vertical":false,"data":[["1","2"],["Antes do QC","Depois do QC"],[5680,1666],[20342.68221830986,24193.98679471789],[11498,25982],[3046.132042253521,4141.81212484994],[2610.5,4640.5]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Etapa<\/th>\n      <th>number_of_cells<\/th>\n      <th>mean_counts_per_cell<\/th>\n      <th>median_counts_per_cell<\/th>\n      <th>mean_features_per_cell<\/th>\n      <th>median_features_per_cell<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,4,5,6]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"Etapa","targets":1},{"name":"number_of_cells","targets":2},{"name":"mean_counts_per_cell","targets":3},{"name":"median_counts_per_cell","targets":4},{"name":"mean_features_per_cell","targets":5},{"name":"median_features_per_cell","targets":6}],"order":[],"autoWidth":false,"orderClasses":false},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
</div>
</section>
<section id="normalização-protocolo-padrão" class="level2 page-columns page-full">
<h2 class="anchored" data-anchor-id="normalização-protocolo-padrão">Normalização — Protocolo padrão</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>seurat_object <span class="ot">&lt;-</span> <span class="fu">NormalizeData</span>(</span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>  seurat_object, <span class="at">normalization.method =</span> <span class="st">"LogNormalize"</span>, <span class="at">scale.factor =</span> <span class="dv">10000</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-5-contents" aria-controls="callout-5" aria-expanded="true" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-5" class="callout-5-contents callout-collapse collapse show">
<div class="callout-body-container callout-body">
<p>A normalização é uma etapa fundamental no pré-processamento de dados de célula única. Ela tem como objetivo <strong>corrigir variações na profundidade de sequenciamento entre células</strong></p>
<p>Métodos disponíveis no argumento <code>normalization.method</code>:</p>
<ul>
<li>“LogNormalize”: método padrão, baseado em normalização por profundidade e log-transformação;</li>
<li>“CLR” (Centered Log-Ratio): útil para dados de proteínas ou marcadores em CITE-seq;</li>
<li>“RC” (Relative Counts): contagens relativas sem transformação logarítmica;</li>
<li>“SCT”: normalização baseada em modelos estatísticos via SCTransform() (mais avançado, recomendado para integração de dados).</li>
</ul>
</div>
</div>
</div>
<section id="encontrando-genes-com-alta-variabilidade" class="level3 page-columns page-full">
<h3 class="anchored" data-anchor-id="encontrando-genes-com-alta-variabilidade">Encontrando genes com alta variabilidade</h3>
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a>seurat_object <span class="ot">&lt;-</span> <span class="fu">FindVariableFeatures</span>(</span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>  seurat_object, <span class="at">selection.method =</span> <span class="st">"vst"</span>, <span class="at">nfeatures =</span> <span class="dv">2000</span>)</span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify the 10 most highly variable genes</span></span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>top10 <span class="ot">&lt;-</span> <span class="fu">head</span>(<span class="fu">VariableFeatures</span>(seurat_object), <span class="dv">10</span>)</span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a><span class="co"># plot variable features with and without labels</span></span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a>plot1 <span class="ot">&lt;-</span> <span class="fu">VariableFeaturePlot</span>(seurat_object)</span>
<span id="cb25-9"><a href="#cb25-9" aria-hidden="true" tabindex="-1"></a>plot2 <span class="ot">&lt;-</span> <span class="fu">LabelPoints</span>(<span class="at">plot =</span> plot1, <span class="at">points =</span> top10, <span class="at">repel =</span> <span class="cn">TRUE</span>)</span>
<span id="cb25-10"><a href="#cb25-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-11"><a href="#cb25-11" aria-hidden="true" tabindex="-1"></a>plot2</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell quarto-layout-panel page-columns page-full" data-layout-nrow="1" data-layout-align="center">
<div class="quarto-layout-row page-full">
<div class="cell-output-display quarto-layout-cell page-columns page-full" style="flex-basis: 100.0%;justify-content: center;">
<div id="fig-highly-variable-genes" class="quarto-float quarto-figure quarto-figure-center anchored page-columns page-full" data-fig-align="center" width="672">
<figure class="quarto-float quarto-float-fig figure page-columns page-full">
<div aria-describedby="fig-highly-variable-genes-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca" class="page-columns page-full">
<a href="03_quality_control_files/figure-html/fig-highly-variable-genes-1.png" class="lightbox page-columns page-full" data-gallery="quarto-lightbox-gallery-5" title="Figure&nbsp;5: "><img src="03_quality_control_files/figure-html/fig-highly-variable-genes-1.png" id="fig-highly-variable-genes" class="img-fluid quarto-figure quarto-figure-center anchored figure-img column-screen-inset-shaded" width="672"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig quarto-uncaptioned" id="fig-highly-variable-genes-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;5
</figcaption>
</figure>
</div>
</div>
</div>
</div>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-6-contents" aria-controls="callout-6" aria-expanded="true" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-6" class="callout-6-contents callout-collapse collapse show">
<div class="callout-body-container callout-body">
<p>O código abaixo identifica os <strong>genes com maior variabilidade de expressão</strong> entre as células, que são mais informativos para análises downstream como <strong>PCA, clusterização e UMAP</strong>.</p>
<ul>
<li><code>selection.method</code>: O método <strong>vst</strong> é baseado em transformação de variância estabilizada (log-linear), ideal para scRNA-seq;</li>
<li><code>nfeatures</code>: quantidade de genes a serem mantidos como “altamente variáveis”.</li>
</ul>
</div>
</div>
</div>
</section>
<section id="ajustando-escala-dos-dados" class="level3">
<h3 class="anchored" data-anchor-id="ajustando-escala-dos-dados">Ajustando escala dos dados</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a>seurat_object <span class="ot">&lt;-</span> <span class="fu">ScaleData</span>(seurat_object)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-7-contents" aria-controls="callout-7" aria-expanded="true" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-7" class="callout-7-contents callout-collapse collapse show">
<div class="callout-body-container callout-body">
<p>A função <code>ScaleData()</code> realiza a <strong>padronização dos dados de expressão gênica</strong>, centralizando os valores (subtraindo a média) e escalando pela variância (dividindo pelo desvio padrão) para cada gene.</p>
<p>Essa etapa é essencial para técnicas como <strong>PCA e clustering</strong>, pois garante que todos os genes contribuam igualmente, independentemente da magnitude de sua expressão.</p>
</div>
</div>
</div>
</section>
</section>
<section id="normalização---sctransform" class="level2">
<h2 class="anchored" data-anchor-id="normalização---sctransform">Normalização - SCTransform</h2>
<p>Já o <strong><code>SCTransform()</code></strong> substitui essas três etapas com um único comando baseado em um <strong>modelo estatístico de regressão negativa binomial</strong>, que:</p>
<ul>
<li>Normaliza os dados levando em conta a variância técnica estimada;</li>
<li>Identifica genes variáveis de forma mais robusta;</li>
<li>Corrige efeitos indesejados (como % de genes mitocondriais ou batch effect) diretamente no modelo;</li>
<li>Retorna dados já escalados e prontos para PCA.</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a>seurat_object <span class="ot">&lt;-</span> <span class="fu">SCTransform</span>(</span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a>  seurat_object, <span class="at">vars.to.regress =</span> <span class="st">"percent.mt"</span>, <span class="at">verbose =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-8-contents" aria-controls="callout-8" aria-expanded="true" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-8" class="callout-8-contents callout-collapse collapse show">
<div class="callout-body-container callout-body">
<p>O argumento <code>vars.to.regress</code> permite <strong>remover o efeito de variáveis indesejadas</strong> — como porcentagem de transcritos mitocondriais (<code>percent.mt</code>), ciclo celular ou batch — durante a modelagem estatística realizada pelo <code>SCTransform()</code>. Para mais detalhes, por favor acesso o <a href="https://satijalab.org/seurat/articles/sctransform_vignette">link</a></p>
</div>
</div>
</div>
</section>
<section id="redução-de-dimensionalidade" class="level2 page-columns page-full">
<h2 class="anchored" data-anchor-id="redução-de-dimensionalidade">Redução de dimensionalidade</h2>
<p>Dados de scRNA-seq possuem <strong>altíssima dimensionalidade</strong>, com milhares de genes medidos por célula. Trabalhar diretamente com todas essas dimensões é:</p>
<ul>
<li><strong>Computacionalmente ineficiente</strong>;</li>
<li><strong>Propenso a ruído</strong> (a maioria dos genes não é informativa para variação biológica relevante);</li>
<li><strong>Inadequado para visualização</strong> e interpretação humana.</li>
</ul>
<p>A redução de dimensionalidade, como <strong>PCA (análise de componentes principais)</strong> ou <strong>UMAP (projeção para espaço não linear)</strong>, nos permite <strong>resumir os dados em eixos principais de variação</strong>, destacando padrões estruturais — como <strong>grupos celulares, trajetórias ou batch effects</strong> — enquanto minimizamos a influência de ruídos técnicos.</p>
<p>Essa etapa é essencial para <strong>visualização</strong>, <strong>clusterização</strong> e <strong>modelagem de trajetórias</strong> em análises de transcriptômica de célula única.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a>seurat_object <span class="ot">&lt;-</span> <span class="fu">RunPCA</span>(</span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a>  seurat_object, <span class="at">assay =</span> <span class="st">"RNA"</span>, <span class="at">reduction.name =</span> <span class="st">"STD_PCA"</span>, <span class="at">features =</span> <span class="fu">VariableFeatures</span>(<span class="at">object =</span> seurat_object))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-9-contents" aria-controls="callout-9" aria-expanded="true" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-9" class="callout-9-contents callout-collapse collapse show">
<div class="callout-body-container callout-body">
<p>O argumento <code>reduction.name</code> permite <strong>nomear explicitamente o resultado da redução de dimensionalidade</strong>, neste caso, da análise de componentes principais (PCA). Por padrão, o Seurat armazena o PCA no slot <code>"pca"</code>, mas ao especificar:</p>
</div>
</div>
</div>
<div class="cell page-columns page-full" data-layout-align="center">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Examine and visualize PCA results a few different ways</span></span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a><span class="fu">VizDimLoadings</span>(seurat_object, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>, <span class="at">reduction =</span> <span class="st">"STD_PCA"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display page-columns page-full">
<div id="fig-pc-loadings" class="quarto-float quarto-figure quarto-figure-center anchored page-columns page-full" data-fig-align="center" width="672">
<figure class="quarto-float quarto-float-fig figure page-columns page-full">
<div aria-describedby="fig-pc-loadings-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca" class="page-columns page-full">
<a href="03_quality_control_files/figure-html/fig-pc-loadings-1.png" class="lightbox page-columns page-full" data-gallery="quarto-lightbox-gallery-6" title="Figure&nbsp;6: "><img src="03_quality_control_files/figure-html/fig-pc-loadings-1.png" id="fig-pc-loadings" class="img-fluid quarto-figure quarto-figure-center anchored figure-img column-screen-inset-shaded" width="672"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig quarto-uncaptioned" id="fig-pc-loadings-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;6
</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="projeção-com-umap" class="level2 page-columns page-full">
<h2 class="anchored" data-anchor-id="projeção-com-umap">Projeção com UMAP</h2>
<p>O <strong>UMAP (Uniform Manifold Approximation and Projection)</strong> é uma técnica de redução de dimensionalidade <strong>não linear</strong> amplamente utilizada para <strong>visualizar dados de scRNA-seq</strong>.</p>
<p>Ele projeta células de um espaço de milhares de genes para <strong>2D ou 3D</strong>, preservando <strong>estruturas locais e globais</strong> dos dados. Isso facilita a identificação visual de <strong>subpopulações celulares, transições fenotípicas</strong> e possíveis <strong>trajetórias biológicas</strong>, sendo especialmente útil após PCA.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a>seurat_object <span class="ot">&lt;-</span> <span class="fu">RunUMAP</span>(</span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a>  seurat_object, <span class="at">reduction =</span> <span class="st">"STD_PCA"</span>, <span class="at">umap.method =</span> <span class="st">"umap-learn"</span>, <span class="at">reduction.name =</span> <span class="st">"STD_UMAP"</span>, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>, <span class="at">verbose =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-10-contents" aria-controls="callout-10" aria-expanded="true" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-10" class="callout-10-contents callout-collapse collapse show">
<div class="callout-body-container callout-body">
<p>O parâmetro <code>umap.method = "umap-learn"</code> especifica que o Seurat deve usar a <strong>implementação original do UMAP em Python</strong>, fornecida pelo pacote <a href="https://github.com/lmcinnes/umap"><code>umap-learn</code></a>.</p>
<p>Essa versão tende a produzir <strong>projeções mais estáveis e refinadas</strong> do que a implementação padrão em R, especialmente em datasets grandes ou com estrutura complexa. Para funcionar, é necessário ter o Python instalado com o pacote <code>umap-learn</code> disponível via <code>reticulate</code>.</p>
</div>
</div>
</div>
<div class="cell page-columns page-full" data-layout-align="center">
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(seurat_object, <span class="at">reduction =</span> <span class="st">"STD_UMAP"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display page-columns page-full">
<div id="fig-umap-plot" class="quarto-float quarto-figure quarto-figure-center anchored page-columns page-full" data-fig-align="center" width="672">
<figure class="quarto-float quarto-float-fig figure page-columns page-full">
<div aria-describedby="fig-umap-plot-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca" class="page-columns page-full">
<a href="03_quality_control_files/figure-html/fig-umap-plot-1.png" class="lightbox page-columns page-full" data-gallery="quarto-lightbox-gallery-7" title="Figure&nbsp;7: "><img src="03_quality_control_files/figure-html/fig-umap-plot-1.png" id="fig-umap-plot" class="img-fluid quarto-figure quarto-figure-center anchored figure-img column-screen-inset-shaded" width="672"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig quarto-uncaptioned" id="fig-umap-plot-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;7
</figcaption>
</figure>
</div>
</div>
</div>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp("https:\/\/lint-edu\.github\.io\/PPGBM0117\.2025\.1");
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">
      &nbsp;
    </div>   
    <div class="nav-footer-center">
<p>Feito com ❤️ usando <a href="https://quarto.org">Quarto</a></p>
</div>
    <div class="nav-footer-right">
      &nbsp;
    </div>
  </div>
</footer>
<script>var lightboxQuarto = GLightbox({"closeEffect":"zoom","descPosition":"bottom","loop":false,"openEffect":"zoom","selector":".lightbox"});
(function() {
  let previousOnload = window.onload;
  window.onload = () => {
    if (previousOnload) {
      previousOnload();
    }
    lightboxQuarto.on('slide_before_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      const href = trigger.getAttribute('href');
      if (href !== null) {
        const imgEl = window.document.querySelector(`a[href="${href}"] img`);
        if (imgEl !== null) {
          const srcAttr = imgEl.getAttribute("src");
          if (srcAttr && srcAttr.startsWith("data:")) {
            slideConfig.href = srcAttr;
          }
        }
      } 
    });
  
    lightboxQuarto.on('slide_after_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(slideNode);
      }
    });
  
  };
  
})();
          </script>




</body></html>