<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.40">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>04 - Redução de dimensionalidade e clusterização – Treinamento em análise de célula única</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting-549806ee2085284f45b00abea8c6df48.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap-859d446df91ed362524820d8142de709.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script src="../../site_libs/quarto-contrib/glightbox/glightbox.min.js"></script>
<link href="../../site_libs/quarto-contrib/glightbox/glightbox.min.css" rel="stylesheet">
<link href="../../site_libs/quarto-contrib/glightbox/lightbox.css" rel="stylesheet">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<link href="../../site_libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet">
<script src="../../site_libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="../../site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="../../site_libs/datatables-binding-0.33/datatables.js"></script>
<script src="../../site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<link href="../../site_libs/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="../../site_libs/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="../../site_libs/dt-core-1.13.6/js/jquery.dataTables.min.js"></script>
<link href="../../site_libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet">
<script src="../../site_libs/crosstalk-1.2.1/js/crosstalk.min.js"></script>


<link rel="stylesheet" href="../../styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../../index.html">
    <span class="navbar-title">Treinamento em análise de célula única</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../index.html"> 
<span class="menu-text">Início</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../about.html"> 
<span class="menu-text">Sobre</span></a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-semana-1" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Semana 1</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-semana-1">    
        <li>
    <a class="dropdown-item" href="../../material/semana1/01_data_collection.html">
 <span class="dropdown-text">Coleta de Dados</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../material/semana1/02_alignment.html">
 <span class="dropdown-text">Alinhamento</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-semana-2" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Semana 2</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-semana-2">    
        <li>
    <a class="dropdown-item" href="../../material/semana2/03_quality_control.html">
 <span class="dropdown-text">Controle de qualidade</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../material/semana2/04_clustering.html">
 <span class="dropdown-text">Clusterização</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-semana-3" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Semana 3</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-semana-3">    
        <li>
    <a class="dropdown-item" href="../../material/semana3/05_cell_annotation.html">
 <span class="dropdown-text">Anotação de células I</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../material/semana3/06_cell_annotation.html">
 <span class="dropdown-text">Anotação de células II</span></a>
  </li>  
    </ul>
  </li>
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#motivação" id="toc-motivação" class="nav-link active" data-scroll-target="#motivação">Motivação</a></li>
  <li><a href="#contextualização" id="toc-contextualização" class="nav-link" data-scroll-target="#contextualização">Contextualização</a></li>
  <li><a href="#unindo-amostras" id="toc-unindo-amostras" class="nav-link" data-scroll-target="#unindo-amostras">Unindo amostras</a>
  <ul class="collapse">
  <li><a href="#salvando-o-objeto-do-seurat" id="toc-salvando-o-objeto-do-seurat" class="nav-link" data-scroll-target="#salvando-o-objeto-do-seurat">Salvando o objeto do Seurat</a></li>
  </ul></li>
  <li><a href="#normalização" id="toc-normalização" class="nav-link" data-scroll-target="#normalização">Normalização</a></li>
  <li><a href="#redução-de-dimensionalidade-pca" id="toc-redução-de-dimensionalidade-pca" class="nav-link" data-scroll-target="#redução-de-dimensionalidade-pca">Redução de dimensionalidade (PCA)</a>
  <ul class="collapse">
  <li><a href="#visualizando-dimensões" id="toc-visualizando-dimensões" class="nav-link" data-scroll-target="#visualizando-dimensões">Visualizando dimensões</a></li>
  <li><a href="#extraindo-informacões-do-pca" id="toc-extraindo-informacões-do-pca" class="nav-link" data-scroll-target="#extraindo-informacões-do-pca">Extraindo informacões do PCA</a></li>
  </ul></li>
  <li><a href="#construção-da-matriz-de-vizinhança" id="toc-construção-da-matriz-de-vizinhança" class="nav-link" data-scroll-target="#construção-da-matriz-de-vizinhança">Construção da matriz de vizinhança</a></li>
  <li><a href="#clusterização" id="toc-clusterização" class="nav-link" data-scroll-target="#clusterização">Clusterização</a></li>
  <li><a href="#projeção-com-umap" id="toc-projeção-com-umap" class="nav-link" data-scroll-target="#projeção-com-umap">Projeção com UMAP</a></li>
  <li><a href="#expressão-diferencial" id="toc-expressão-diferencial" class="nav-link" data-scroll-target="#expressão-diferencial">Expressão diferencial</a>
  <ul class="collapse">
  <li><a href="#inspecionando-degs" id="toc-inspecionando-degs" class="nav-link" data-scroll-target="#inspecionando-degs">Inspecionando DEGs</a></li>
  </ul></li>
  <li><a href="#salvando-o-objeto-anotado" id="toc-salvando-o-objeto-anotado" class="nav-link" data-scroll-target="#salvando-o-objeto-anotado">Salvando o objeto anotado</a></li>
  <li><a href="#analise-de-pureza-dos-clusters" id="toc-analise-de-pureza-dos-clusters" class="nav-link" data-scroll-target="#analise-de-pureza-dos-clusters">Analise de pureza dos clusters</a>
  <ul class="collapse">
  <li><a href="#calculando-entropia" id="toc-calculando-entropia" class="nav-link" data-scroll-target="#calculando-entropia">Calculando entropia</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content page-columns page-full" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">04 - Redução de dimensionalidade e clusterização</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<section id="motivação" class="level2">
<h2 class="anchored" data-anchor-id="motivação">Motivação</h2>
<p>Em experimentos de transcriptômica de célula única (scRNA-seq), a alta dimensionalidade dos dados — frequentemente composta por milhares de genes — dificulta a análise direta e a interpretação biológica. Para identificar populações celulares de forma robusta, é necessário reduzir essa dimensionalidade preservando as principais variações biológicas. Posteriormente, técnicas de clusterização são empregadas para agrupar células com perfis transcriptômicos semelhantes, revelando estruturas ocultas nos dados, como tipos celulares ou estados funcionais distintos.</p>
<p>Este tutorial descreve as etapas fundamentais de <strong>união de amostras</strong>, <strong>redução de dimensionalidade</strong>, <strong>clusterização</strong> e <strong>subclusterização</strong> de células, utilizando o pacote <strong>Seurat</strong>.</p>
</section>
<section id="contextualização" class="level2">
<h2 class="anchored" data-anchor-id="contextualização">Contextualização</h2>
<p>Neste tutorial, trabalharemos com objetos do Seurat e utilizaremos as seguintes funções principais:</p>
<ul>
<li><code>merge()</code> — para unir objetos de diferentes amostras.</li>
<li><code>SCTransform()</code> ou <code>NormalizeData()</code> — para normalização e modelagem da variação técnica.</li>
<li><code>RunPCA()</code> — para a redução inicial de dimensionalidade via Análise de Componentes Principais (PCA).</li>
<li><code>FindNeighbors()</code> e <code>FindClusters()</code> — para identificar agrupamentos celulares.</li>
<li><code>RunUMAP()</code> ou <code>RunTSNE()</code> — para projeção em duas dimensões, facilitando a visualização.</li>
<li><code>Subset()</code> — para subclusterizar grupos de interesse.</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Seurat)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ROGUE)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(presto)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(here)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Definindo diretorio padrao</span></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>opts_knit<span class="sc">$</span><span class="fu">set</span>(</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">root.dir =</span> <span class="st">"/home/oandrefonseca/Disciplinas/PPGBM0117.2025.1"</span>,</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">verbose =</span> <span class="cn">FALSE</span></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Aumentar o limite de uso de memoria</span></span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">future.globals.maxSize =</span> <span class="dv">5</span> <span class="sc">*</span> <span class="dv">1024</span><span class="sc">^</span><span class="dv">3</span>) <span class="co"># 5 GB, por exemplo</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="unindo-amostras" class="level2">
<h2 class="anchored" data-anchor-id="unindo-amostras">Unindo amostras</h2>
<p>Antes de qualquer redução de dimensionalidade, é necessário unir diferentes amostras em um único objeto Seurat. Para isso, usamos a função <code>merge()</code>. Abaixo um exemplo de codigo de realizar a juncao.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Listar os arquivos RDS</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>files <span class="ot">&lt;-</span> <span class="fu">list.files</span>(</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">path =</span> <span class="st">"path/to/your/objects"</span>, <span class="at">pattern =</span> <span class="st">"</span><span class="sc">\\</span><span class="st">.rds$"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Carregar todos os objetos em uma lista</span></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>seurat_list <span class="ot">&lt;-</span> <span class="fu">lapply</span>(files, readRDS)</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Criar identificadores a partir dos nomes dos arquivos</span></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>sample_ids <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">.rds$"</span>, <span class="st">""</span>, <span class="fu">basename</span>(files))</span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Fazer o merge</span></span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a>seurat_merged <span class="ot">&lt;-</span> <span class="fu">merge</span>(</span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a>  <span class="at">x =</span> seurat_list[[<span class="dv">1</span>]],</span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a>  <span class="at">y =</span> seurat_list[<span class="sc">-</span><span class="dv">1</span>],</span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a>  <span class="at">add.cell.ids =</span> sample_ids,</span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a>  <span class="at">project =</span> <span class="st">"ProjectName"</span></span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="salvando-o-objeto-do-seurat" class="level3">
<h3 class="anchored" data-anchor-id="salvando-o-objeto-do-seurat">Salvando o objeto do Seurat</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">file.exists</span>(<span class="st">"./data/seurat_object.RDS"</span>)) {</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">saveRDS</span>(<span class="at">file =</span> <span class="st">"./data/seurat_object.RDS"</span>, <span class="at">object =</span> seurat_merged)</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Objeto contendo todas as amostras</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>seurat_object <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"./data/seurat_object.RDS"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="normalização" class="level2">
<h2 class="anchored" data-anchor-id="normalização">Normalização</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>seurat_object <span class="ot">&lt;-</span> <span class="fu">NormalizeData</span>(</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>  seurat_object, <span class="at">normalization.method =</span> <span class="st">"LogNormalize"</span>, <span class="at">scale.factor =</span> <span class="dv">10000</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Normalizing layer: counts</code></pre>
</div>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>seurat_object <span class="ot">&lt;-</span> <span class="fu">FindVariableFeatures</span>(</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>  seurat_object, <span class="at">selection.method =</span> <span class="st">"vst"</span>, <span class="at">nfeatures =</span> <span class="dv">2000</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Finding variable features for layer counts</code></pre>
</div>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>seurat_object <span class="ot">&lt;-</span> <span class="fu">ScaleData</span>(</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>  seurat_object, <span class="at">features =</span> <span class="fu">VariableFeatures</span>(seurat_object))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Centering and scaling data matrix</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Different features in new layer data than already exists for
scale.data</code></pre>
</div>
</div>
</section>
<section id="redução-de-dimensionalidade-pca" class="level2 page-columns page-full">
<h2 class="anchored" data-anchor-id="redução-de-dimensionalidade-pca">Redução de dimensionalidade (PCA)</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>seurat_object <span class="ot">&lt;-</span> <span class="fu">RunPCA</span>(seurat_object, <span class="at">npcs =</span> <span class="dv">50</span>, <span class="at">verbose =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ElbowPlot</span>(seurat_object, <span class="at">reduction =</span> <span class="st">"pca"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell quarto-layout-panel page-columns page-full" data-layout-align="center" data-layout-nrow="1">
<div class="quarto-layout-row page-full">
<div class="cell-output-display quarto-layout-cell page-columns page-full" style="flex-basis: 100.0%;justify-content: center;">
<div id="fig-elbow" class="quarto-float quarto-figure quarto-figure-center anchored page-columns page-full" data-fig-align="center" width="672">
<figure class="quarto-float quarto-float-fig figure page-columns page-full">
<div aria-describedby="fig-elbow-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca" class="page-columns page-full">
<a href="04_clustering_files/figure-html/fig-elbow-1.png" class="lightbox page-columns page-full" data-gallery="quarto-lightbox-gallery-1" title="Figure&nbsp;1: "><img src="04_clustering_files/figure-html/fig-elbow-1.png" id="fig-elbow" class="img-fluid quarto-figure quarto-figure-center anchored figure-img column-screen-inset-shaded" width="672"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig quarto-uncaptioned" id="fig-elbow-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;1
</figcaption>
</figure>
</div>
</div>
</div>
</div>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-1-contents" aria-controls="callout-1" aria-expanded="true" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-1" class="callout-1-contents callout-collapse collapse show">
<div class="callout-body-container callout-body">
<p>O <code>ElbowPlot(seurat_object)</code> é utilizado para identificar o número apropriado de componentes principais (PCs) a serem considerados nas análises downstream, como clusterização e redução de dimensionalidade. O gráfico exibe a variação explicada por cada PC; o ponto onde há uma inflexão (“cotovelo”) indica que PCs adicionais contribuem pouco para a variabilidade dos dados. Selecionar o número correto de PCs é essencial para capturar a estrutura biológica relevante, minimizando ruído técnico.</p>
</div>
</div>
</div>
<section id="visualizando-dimensões" class="level3 page-columns page-full">
<h3 class="anchored" data-anchor-id="visualizando-dimensões">Visualizando dimensões</h3>
<div class="cell page-columns page-full" data-layout-align="center">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Examine and visualize PCA results a few different ways</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a><span class="fu">VizDimLoadings</span>(seurat_object, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>, <span class="at">reduction =</span> <span class="st">"pca"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display page-columns page-full">
<div id="fig-pc-loadings" class="quarto-float quarto-figure quarto-figure-center anchored page-columns page-full" data-fig-align="center" width="672">
<figure class="quarto-float quarto-float-fig figure page-columns page-full">
<div aria-describedby="fig-pc-loadings-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca" class="page-columns page-full">
<a href="04_clustering_files/figure-html/fig-pc-loadings-1.png" class="lightbox page-columns page-full" data-gallery="quarto-lightbox-gallery-2" title="Figure&nbsp;2: "><img src="04_clustering_files/figure-html/fig-pc-loadings-1.png" id="fig-pc-loadings" class="img-fluid quarto-figure quarto-figure-center anchored figure-img column-screen-inset-shaded" width="672"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig quarto-uncaptioned" id="fig-pc-loadings-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;2
</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="extraindo-informacões-do-pca" class="level3">
<h3 class="anchored" data-anchor-id="extraindo-informacões-do-pca">Extraindo informacões do PCA</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Get cell embeddings and feature loadings stored on pbmc[['pca']]@cell.embeddings</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="fu">Embeddings</span>(seurat_object, <span class="at">reduction =</span> <span class="st">"pca"</span>)</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a><span class="co"># stored in pbmc[['pca]]@feature.loadings</span></span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a><span class="fu">Loadings</span>(seurat_object, <span class="at">reduction =</span> <span class="st">"pca"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="construção-da-matriz-de-vizinhança" class="level2">
<h2 class="anchored" data-anchor-id="construção-da-matriz-de-vizinhança">Construção da matriz de vizinhança</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>seurat_object <span class="ot">&lt;-</span> <span class="fu">FindNeighbors</span>(seurat_object, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Computing nearest neighbor graph</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Computing SNN</code></pre>
</div>
</div>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-2-contents" aria-controls="callout-2" aria-expanded="true" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-2" class="callout-2-contents callout-collapse collapse show">
<div class="callout-body-container callout-body">
<p>A função <code>FindNeighbors</code> constrói uma matriz de vizinhança baseada nos embeddings de redução de dimensionalidade (tipicamente PCA). Cada célula é conectada às suas células mais próximas no espaço de baixa dimensão, formando um <strong>grafo de similaridade</strong>.</p>
<p>Esse grafo é o alicerce para a clusterização subsequente (<code>FindClusters()</code>), permitindo identificar comunidades de células com perfis transcriptômicos semelhantes. O parâmetro <code>dims</code> especifica quais componentes principais serão usados para calcular as distâncias entre as células.</p>
</div>
</div>
</div>
</section>
<section id="clusterização" class="level2">
<h2 class="anchored" data-anchor-id="clusterização">Clusterização</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>seurat_object <span class="ot">&lt;-</span> <span class="fu">FindClusters</span>(</span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>  seurat_object, <span class="at">algorithm =</span> <span class="dv">4</span>, <span class="at">resolution =</span> <span class="fu">c</span>(<span class="fl">0.25</span>, <span class="fl">0.5</span>, <span class="fl">1.0</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-3-contents" aria-controls="callout-3" aria-expanded="true" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-3" class="callout-3-contents callout-collapse collapse show">
<div class="callout-body-container callout-body">
<p>A função <code>FindClusters</code> realiza a clusterização das células com base na matriz de vizinhança previamente calculada.<br>
Ela aplica algoritmos de otimização de comunidades (como o algoritmo Louvain ou Leiden) para agrupar células com perfis transcriptômicos semelhantes.</p>
<p>O parâmetro <code>resolution</code> controla a <strong>granularidade</strong> da clusterização, por exemplo: - <strong>Valores mais baixos</strong> (<code>&lt; 0.5</code>) resultam em <strong>menos clusters</strong> (agrupamentos maiores). - <strong>Valores mais altos</strong> (<code>&gt; 0.5</code>) resultam em <strong>mais clusters</strong> (agrupamentos menores e mais específicos).</p>
<p>A escolha do <code>resolution</code> deve ser guiada tanto por critérios biológicos quanto por inspeção visual dos agrupamentos gerados.</p>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Explore as resolutions</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>DT<span class="sc">::</span><span class="fu">datatable</span>(<span class="fu">head</span>(seurat_object<span class="sc">@</span>meta.data, <span class="at">n =</span> <span class="dv">100</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="datatables html-widget html-fill-item" id="htmlwidget-530cfa52d3babb35245b" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-530cfa52d3babb35245b">{"x":{"filter":"none","vertical":false,"data":[["SRR8315733_AAACCTGCAGATCGGA-1","SRR8315733_AAACGGGCATAGACTC-1","SRR8315733_AAACGGGTCATACGGT-1","SRR8315733_AAACGGGTCATAGCAC-1","SRR8315733_AAACGGGTCTTGGGTA-1","SRR8315733_AAAGATGAGACAGGCT-1","SRR8315733_AAAGATGCACAAGCCC-1","SRR8315733_AAAGATGGTTTGGGCC-1","SRR8315733_AAAGATGTCTGAGTGT-1","SRR8315733_AAAGCAAAGAGTTGGC-1","SRR8315733_AAAGCAAAGATAGCAT-1","SRR8315733_AAAGCAAAGCAGGTCA-1","SRR8315733_AAAGCAAAGCCACGCT-1","SRR8315733_AAAGTAGGTATATGAG-1","SRR8315733_AAAGTAGGTCTGCCAG-1","SRR8315733_AAATGCCAGACTGTAA-1","SRR8315733_AAATGCCCATGTTGAC-1","SRR8315733_AAATGCCGTAACGACG-1","SRR8315733_AAATGCCTCCTCTAGC-1","SRR8315733_AAATGCCTCGTTTGCC-1","SRR8315733_AAATGCCTCTCTTGAT-1","SRR8315733_AACACGTAGCCACGTC-1","SRR8315733_AACCATGCACTTGGAT-1","SRR8315733_AACCATGCATTGAGCT-1","SRR8315733_AACCATGGTTATCGGT-1","SRR8315733_AACCGCGAGGTGTTAA-1","SRR8315733_AACCGCGCACGAAAGC-1","SRR8315733_AACCGCGGTCTTCAAG-1","SRR8315733_AACGTTGAGGTAGCCA-1","SRR8315733_AACGTTGAGTACATGA-1","SRR8315733_AACGTTGCACATCCGG-1","SRR8315733_AACTCAGCATTTCAGG-1","SRR8315733_AACTCAGGTGGCGAAT-1","SRR8315733_AACTCAGTCCCGACTT-1","SRR8315733_AACTCAGTCTCCTATA-1","SRR8315733_AACTCAGTCTCGCTTG-1","SRR8315733_AACTCAGTCTTAGAGC-1","SRR8315733_AACTCCCAGTGGGATC-1","SRR8315733_AACTCCCTCAGCTTAG-1","SRR8315733_AACTCCCTCCTGCTTG-1","SRR8315733_AACTCTTAGAAGCCCA-1","SRR8315733_AACTCTTGTGGTGTAG-1","SRR8315733_AACTCTTTCCGTCATC-1","SRR8315733_AACTGGTAGACCACGA-1","SRR8315733_AACTGGTCACGTCTCT-1","SRR8315733_AACTGGTGTCCGCTGA-1","SRR8315733_AACTGGTTCAACCATG-1","SRR8315733_AACTGGTTCCCGACTT-1","SRR8315733_AACTTTCAGACACTAA-1","SRR8315733_AACTTTCCAGTAAGCG-1","SRR8315733_AAGACCTAGGGATCTG-1","SRR8315733_AAGACCTCATAACCTG-1","SRR8315733_AAGACCTGTATGAATG-1","SRR8315733_AAGACCTGTTCAGGCC-1","SRR8315733_AAGCCGCCAGTATAAG-1","SRR8315733_AAGCCGCGTCTAGAGG-1","SRR8315733_AAGCCGCGTCTGATCA-1","SRR8315733_AAGCCGCTCGTCACGG-1","SRR8315733_AAGGAGCAGGAGTCTG-1","SRR8315733_AAGGAGCCATACAGCT-1","SRR8315733_AAGGCAGAGGATGCGT-1","SRR8315733_AAGGCAGCAAGCTGAG-1","SRR8315733_AAGGTTCCATCTCGCT-1","SRR8315733_AAGTCTGCAGCCTTTC-1","SRR8315733_AAGTCTGCATTCTTAC-1","SRR8315733_AATCCAGAGATGTTAG-1","SRR8315733_ACACCCTGTACAGCAG-1","SRR8315733_ACACCCTTCAACACCA-1","SRR8315733_ACACCCTTCAGCTCTC-1","SRR8315733_ACACCCTTCATCTGCC-1","SRR8315733_ACACCCTTCCCGACTT-1","SRR8315733_ACACCCTTCTTCATGT-1","SRR8315733_ACACCGGTCAATACCG-1","SRR8315733_ACACCGGTCGCGTAGC-1","SRR8315733_ACACTGAAGGGTGTTG-1","SRR8315733_ACACTGACACGGTGTC-1","SRR8315733_ACACTGACAGACAAAT-1","SRR8315733_ACACTGATCTTCGGTC-1","SRR8315733_ACAGCCGCAACGATGG-1","SRR8315733_ACAGCCGGTACTTAGC-1","SRR8315733_ACAGCCGGTTCCACTC-1","SRR8315733_ACAGCTAAGGTGGGTT-1","SRR8315733_ACAGCTAAGTGAAGTT-1","SRR8315733_ACATACGTCCTAGGGC-1","SRR8315733_ACCAGTAAGCTGTTCA-1","SRR8315733_ACCAGTAAGTCAAGCG-1","SRR8315733_ACCAGTAGTGACTACT-1","SRR8315733_ACCAGTATCACCACCT-1","SRR8315733_ACCCACTTCCGCAAGC-1","SRR8315733_ACCGTAACAGCTGTGC-1","SRR8315733_ACCGTAAGTGTATGGG-1","SRR8315733_ACCGTAATCTGCTGCT-1","SRR8315733_ACCTTTAAGAGGTACC-1","SRR8315733_ACCTTTACATCACAAC-1","SRR8315733_ACGAGCCAGATATGCA-1","SRR8315733_ACGAGCCAGTCTCAAC-1","SRR8315733_ACGAGCCCAGTACACT-1","SRR8315733_ACGAGCCTCCACGAAT-1","SRR8315733_ACGAGGACAGGACCCT-1","SRR8315733_ACGATACGTGCAACGA-1"],["SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733"],[3211,2655,3064,4757,1802,5048,2542,5812,6049,2748,1009,4832,2838,3730,9037,4722,2240,6301,2659,2689,3232,3465,7898,2818,5323,5196,6640,1068,6440,3918,3755,2145,2557,2783,2819,5369,3570,4480,2952,4361,5203,6443,3195,4364,4122,4508,5100,10136,1700,3945,4775,1053,3241,4002,4332,4928,5645,1270,6229,3407,7870,3778,6744,4867,3584,4738,10181,5238,4112,2715,5498,2886,3972,2236,3250,2195,5601,5215,1978,5714,4679,11599,4348,3055,2977,3802,7999,3975,4592,7391,1772,3832,2567,4350,7970,4367,1446,4304,3154,3763],[1220,1687,1551,1895,718,2285,1164,2489,2477,1304,682,2134,1509,1754,2977,2033,1313,2273,1403,1306,1812,1615,2944,1669,1764,2055,2082,714,1896,1734,1774,896,1123,1379,1305,1934,1674,1620,1375,1844,1519,1751,1428,1995,1764,2109,1673,2227,1166,1167,1960,636,1543,1529,2373,1729,1791,809,2302,1517,2996,1936,2222,1992,1670,1965,3456,2063,1312,1306,2348,1555,1390,1100,1714,1380,2206,2055,1168,2162,1700,2495,1717,1599,1600,1459,2595,2219,1773,2823,876,1537,920,1794,2268,1798,616,1564,1520,1515],[6.197446278417938,2.071563088512241,3.753263707571802,4.729871767920959,5.382907880133185,3.347860538827258,5.704169944925256,6.727460426703373,2.793850223177385,3.966521106259097,18.53320118929634,3.82864238410596,5.14446793516561,2.493297587131367,0.5643465751908819,3.113087674714104,4.241071428571429,3.554991271226789,2.218879277924032,5.132019338043882,3.155940594059406,5.800865800865801,5.58369207394277,0,3.419124553823032,2.925327174749807,3.418674698795181,4.119850187265917,4.37888198757764,6.125574272588056,5.272969374167777,7.832167832167833,6.022682831443097,4.132231404958678,5.143667967364314,2.253678524864966,6.890756302521009,2.723214285714286,3.387533875338753,3.095620270580142,3.075148952527388,3.197268353251591,5.13302034428795,1.993583868010999,4.027171276079573,2.839396628216504,2.431372549019608,3.107734806629834,5.11764705882353,6.210392902408111,3.413612565445026,22.12725546058879,6.880592409750077,3.373313343328336,2.377654662973222,5.174512987012987,2.83436669619132,3.700787401574803,4.527211430406164,5.019078368065747,4.434561626429479,4.870301746956061,3.603202846975089,2.568317238545305,3.571428571428571,4.432249894470241,3.064531971319124,2.176403207331042,2.821011673151751,2.83609576427256,1.382320843943252,5.266805266805267,5.740181268882175,2.459749552772809,0.5846153846153846,1.412300683371299,2.338868059275129,3.048897411313519,1.66835187057634,3.955197759887994,3.269929472109425,3.034744374515044,4.806807727690892,5.368248772504092,5.005038629492778,6.759600210415571,3.050381297662208,3.471698113207547,2.221254355400697,1.988905425517521,4.909706546275395,4.07098121085595,3.389170237631477,2.436781609195402,2.672521957340025,3.778337531486146,6.085753803596127,3.62453531598513,3.26569435637286,4.783417486048366],["GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735"],["SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733","SRR8315733"],["RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq","RNA-Seq"],[124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124,124],[8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724,8274414724],["PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910","PRJNA509910"],["SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740","SAMN10589740"],[3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396,3765245396],["CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+","CD45+ CD3+"],["su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001","su001"],["SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478","SRX5128478"],["GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735","GSM3511735"],["Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000","Illumina HiSeq 4000"],["PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED","PAIRED"],["cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA","cDNA"],["TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC","TRANSCRIPTOMIC"],["Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens","Homo sapiens"],["ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA","ILLUMINA"],["Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor","Basal cell carinoma tumor"],["SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389","SRP173389"],["pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre","pre"],["0","9","1","1","1","1","3","1","1","0","3","3","3","3","3","1","3","1","3","1","3","3","1","3","0","3","1","3","1","0","3","1","3","1","1","1","7","4","4","1","1","1","1","1","1","0","1","1","1","1","1","1","1","3","1","1","1","3","1","3","1","1","1","3","3","1","3","3","3","3","3","3","1","3","3","3","1","1","3","1","1","1","0","3","0","3","1","3","1","3","3","3","1","3","1","1","1","1","0","1"],["5","8","4","5","8","3","5","16","5","1","8","3","3","3","3","5","3","5","3","5","3","5","7","3","1","3","10","8","5","1","12","8","5","5","5","5","18","10","5","5","10","5","1","10","5","10","5","5","5","8","5","10","5","3","3","10","5","5","10","3","5","3","10","3","3","5","3","3","3","1","3","3","5","3","5","10","5","5","3","5","5","5","1","3","1","3","5","5","5","7","5","5","5","3","5","5","8","10","1","5"],["2","2","1","2","2","2","2","10","2","1","2","2","3","3","3","2","2","2","3","2","3","2","3","3","1","3","2","2","2","1","3","2","2","2","2","2","1","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","3","3","2","2","2","2","3","2","3","2","2","3","2","3","3","3","2","3","2","2","3","2","2","2","2","3","2","2","2","1","3","1","3","2","2","2","3","2","2","2","3","2","2","2","2","1","2"],["3","8","5","3","8","3","3","11","3","2","8","3","1","1","1","3","3","3","1","3","1","3","1","1","2","1","3","8","3","2","1","8","3","3","3","3","14","3","3","3","3","3","2","3","3","3","3","3","3","3","3","3","3","1","3","3","3","3","3","1","3","1","3","3","1","3","1","1","1","2","1","3","3","1","3","3","3","3","1","3","3","3","2","1","2","1","3","3","3","1","3","3","3","1","3","3","8","3","2","3"],["5","8","4","5","8","3","5","16","5","1","8","3","3","3","3","5","3","5","3","5","3","5","7","3","1","3","10","8","5","1","12","8","5","5","5","5","18","10","5","5","10","5","1","10","5","10","5","5","5","8","5","10","5","3","3","10","5","5","10","3","5","3","10","3","3","5","3","3","3","1","3","3","5","3","5","10","5","5","3","5","5","5","1","3","1","3","5","5","5","7","5","5","5","3","5","5","8","10","1","5"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>orig.ident<\/th>\n      <th>nCount_RNA<\/th>\n      <th>nFeature_RNA<\/th>\n      <th>percent_mito<\/th>\n      <th>Sample.Name<\/th>\n      <th>Run<\/th>\n      <th>Assay.Type<\/th>\n      <th>AvgSpotLen<\/th>\n      <th>Bases<\/th>\n      <th>BioProject<\/th>\n      <th>BioSample<\/th>\n      <th>Bytes<\/th>\n      <th>cell_type<\/th>\n      <th>Donor<\/th>\n      <th>Experiment<\/th>\n      <th>GEO_Accession..exp.<\/th>\n      <th>Instrument<\/th>\n      <th>LibraryLayout<\/th>\n      <th>LibrarySelection<\/th>\n      <th>LibrarySource<\/th>\n      <th>Organism<\/th>\n      <th>Platform<\/th>\n      <th>source_name<\/th>\n      <th>SRA.Study<\/th>\n      <th>treatment<\/th>\n      <th>snn_npc50_res.0.25<\/th>\n      <th>seurat_clusters<\/th>\n      <th>RNA_snn_res.0.25<\/th>\n      <th>RNA_snn_res.0.5<\/th>\n      <th>RNA_snn_res.1<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,4,8,9,12]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"orig.ident","targets":1},{"name":"nCount_RNA","targets":2},{"name":"nFeature_RNA","targets":3},{"name":"percent_mito","targets":4},{"name":"Sample.Name","targets":5},{"name":"Run","targets":6},{"name":"Assay.Type","targets":7},{"name":"AvgSpotLen","targets":8},{"name":"Bases","targets":9},{"name":"BioProject","targets":10},{"name":"BioSample","targets":11},{"name":"Bytes","targets":12},{"name":"cell_type","targets":13},{"name":"Donor","targets":14},{"name":"Experiment","targets":15},{"name":"GEO_Accession..exp.","targets":16},{"name":"Instrument","targets":17},{"name":"LibraryLayout","targets":18},{"name":"LibrarySelection","targets":19},{"name":"LibrarySource","targets":20},{"name":"Organism","targets":21},{"name":"Platform","targets":22},{"name":"source_name","targets":23},{"name":"SRA.Study","targets":24},{"name":"treatment","targets":25},{"name":"snn_npc50_res.0.25","targets":26},{"name":"seurat_clusters","targets":27},{"name":"RNA_snn_res.0.25","targets":28},{"name":"RNA_snn_res.0.5","targets":29},{"name":"RNA_snn_res.1","targets":30}],"order":[],"autoWidth":false,"orderClasses":false},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
</section>
<section id="projeção-com-umap" class="level2 page-columns page-full">
<h2 class="anchored" data-anchor-id="projeção-com-umap">Projeção com UMAP</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>seurat_object <span class="ot">&lt;-</span> <span class="fu">RunUMAP</span>(seurat_object, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: The default method for RunUMAP has changed from calling Python UMAP via reticulate to the R-native UWOT using the cosine metric
To use Python UMAP via reticulate, set umap.method to 'umap-learn' and metric to 'correlation'
This message will be shown once per session</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>10:22:10 UMAP embedding parameters a = 0.9922 b = 1.112</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>10:22:10 Read 67633 rows and found 20 numeric columns</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>10:22:10 Using Annoy for neighbor search, n_neighbors = 30</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>10:22:10 Building Annoy index with metric = cosine, n_trees = 50</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>0%   10   20   30   40   50   60   70   80   90   100%</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>[----|----|----|----|----|----|----|----|----|----|</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>**************************************************|
10:22:16 Writing NN index file to temp file /tmp/Rtmp4AJWOy/file117418fdd87b
10:22:16 Searching Annoy index using 1 thread, search_k = 3000
10:22:40 Annoy recall = 100%
10:22:40 Commencing smooth kNN distance calibration using 1 thread with target n_neighbors = 30
10:22:42 Initializing from normalized Laplacian + noise (using RSpectra)
10:23:00 Commencing optimization for 200 epochs, with 3039738 positive edges
10:23:29 Optimization finished</code></pre>
</div>
</div>
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Por padrao apenas os clusters da ultima resolucao sao apresentados</span></span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(seurat_object, <span class="at">reduction =</span> <span class="st">"umap"</span>, <span class="at">label =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell quarto-layout-panel page-columns page-full" data-layout-align="center" data-layout-nrow="1">
<div class="quarto-layout-row page-full">
<div class="cell-output-display quarto-layout-cell page-columns page-full" style="flex-basis: 100.0%;justify-content: center;">
<div id="fig-umap" class="quarto-float quarto-figure quarto-figure-center anchored page-columns page-full" data-fig-align="center" width="672">
<figure class="quarto-float quarto-float-fig figure page-columns page-full">
<div aria-describedby="fig-umap-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca" class="page-columns page-full">
<a href="04_clustering_files/figure-html/fig-umap-1.png" class="lightbox page-columns page-full" data-gallery="quarto-lightbox-gallery-3" title="Figure&nbsp;3: "><img src="04_clustering_files/figure-html/fig-umap-1.png" id="fig-umap" class="img-fluid quarto-figure quarto-figure-center anchored figure-img column-screen-inset-shaded" width="672"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig quarto-uncaptioned" id="fig-umap-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3
</figcaption>
</figure>
</div>
</div>
</div>
</div>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-4-contents" aria-controls="callout-4" aria-expanded="true" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-4" class="callout-4-contents callout-collapse collapse show">
<div class="callout-body-container callout-body">
<p><strong>Importante:</strong> A clusterização de scRNA-seq <strong>não é um processo exato</strong>. Ela deve ser encarada como uma <strong>ferramenta de geração de hipóteses</strong>. Sempre valide biologicamente seus agrupamentos antes de tirar conclusões definitivas.</p>
</div>
</div>
</div>
</section>
<section id="expressão-diferencial" class="level2">
<h2 class="anchored" data-anchor-id="expressão-diferencial">Expressão diferencial</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="co"># O parametro `max.cells.per.ident` foi acrescentado para otimizar o tempo computacional. Remove-o para analises reais.</span></span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>cluster_markers <span class="ot">&lt;-</span> <span class="fu">FindAllMarkers</span>(seurat_object, <span class="at">max.cells.per.ident =</span> <span class="dv">1000</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="inspecionando-degs" class="level3">
<h3 class="anchored" data-anchor-id="inspecionando-degs">Inspecionando DEGs</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Removendo resultados nao significativos</span></span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a>cluster_markers_top <span class="ot">&lt;-</span> cluster_markers <span class="sc">%&gt;%</span></span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_val_adj <span class="sc">&lt;=</span> <span class="fl">0.01</span>) <span class="sc">%&gt;%</span></span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(cluster) <span class="sc">%&gt;%</span></span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">top_n</span>(<span class="dv">20</span>, avg_log2FC)</span>
<span id="cb32-6"><a href="#cb32-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-7"><a href="#cb32-7" aria-hidden="true" tabindex="-1"></a>DT<span class="sc">::</span><span class="fu">datatable</span>(cluster_markers_top)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="salvando-o-objeto-anotado" class="level2">
<h2 class="anchored" data-anchor-id="salvando-o-objeto-anotado">Salvando o objeto anotado</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(seurat_object, <span class="at">file =</span> <span class="st">"data/seurat_clustered.RDS"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="analise-de-pureza-dos-clusters" class="level2 page-columns page-full">
<h2 class="anchored" data-anchor-id="analise-de-pureza-dos-clusters">Analise de pureza dos clusters</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a>downsampled_object <span class="ot">&lt;-</span> <span class="fu">subset</span>(<span class="at">x =</span> seurat_object, <span class="at">downsample =</span> <span class="dv">1000</span>)</span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a>counts_matrix <span class="ot">&lt;-</span> <span class="fu">LayerData</span>(</span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a>  downsampled_object, <span class="at">assay =</span> <span class="st">"RNA"</span>, <span class="at">layer =</span> <span class="st">"counts"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="calculando-entropia" class="level3 page-columns page-full">
<h3 class="anchored" data-anchor-id="calculando-entropia">Calculando entropia</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a>entropy_calculation <span class="ot">&lt;-</span> <span class="fu">SE_fun</span>(counts_matrix)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a>rogue_value <span class="ot">&lt;-</span> <span class="fu">CalculateRogue</span>(</span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a>  entropy_calculation, </span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">platform =</span> <span class="st">"UMI"</span>)</span>
<span id="cb36-4"><a href="#cb36-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-5"><a href="#cb36-5" aria-hidden="true" tabindex="-1"></a>rogue_value</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.2160008</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a>rogue_result <span class="ot">&lt;-</span> <span class="fu">rogue</span>(</span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a>  counts_matrix, </span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">labels =</span> downsampled_object<span class="sc">@</span>meta.data<span class="sc">$</span>seurat_clusters, </span>
<span id="cb38-4"><a href="#cb38-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">samples =</span> downsampled_object<span class="sc">@</span>meta.data<span class="sc">$</span>orig.ident, </span>
<span id="cb38-5"><a href="#cb38-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">platform =</span> <span class="st">"UMI"</span>, </span>
<span id="cb38-6"><a href="#cb38-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">span =</span> <span class="fl">0.6</span></span>
<span id="cb38-7"><a href="#cb38-7" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rogue.boxplot</span>(rogue_result)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell quarto-layout-panel page-columns page-full" data-layout-align="center" data-layout-nrow="1">
<div class="quarto-layout-row page-full">
<div class="cell-output-display quarto-layout-cell page-columns page-full" style="flex-basis: 100.0%;justify-content: center;">
<div id="fig-rogue" class="quarto-float quarto-figure quarto-figure-center anchored page-columns page-full" data-fig-align="center" width="672">
<figure class="quarto-float quarto-float-fig figure page-columns page-full">
<div aria-describedby="fig-rogue-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca" class="page-columns page-full">
<a href="04_clustering_files/figure-html/fig-rogue-1.png" class="lightbox page-columns page-full" data-gallery="quarto-lightbox-gallery-4" title="Figure&nbsp;4: "><img src="04_clustering_files/figure-html/fig-rogue-1.png" id="fig-rogue" class="img-fluid quarto-figure quarto-figure-center anchored figure-img column-screen-inset-shaded" width="672"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig quarto-uncaptioned" id="fig-rogue-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;4
</figcaption>
</figure>
</div>
</div>
</div>
</div>


</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp("https:\/\/lint-edu\.github\.io\/PPGBM0117\.2025\.1");
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">
      &nbsp;
    </div>   
    <div class="nav-footer-center">
<p>Feito com ❤️ usando <a href="https://quarto.org">Quarto</a></p>
</div>
    <div class="nav-footer-right">
      &nbsp;
    </div>
  </div>
</footer>
<script>var lightboxQuarto = GLightbox({"closeEffect":"zoom","descPosition":"bottom","loop":false,"openEffect":"zoom","selector":".lightbox"});
(function() {
  let previousOnload = window.onload;
  window.onload = () => {
    if (previousOnload) {
      previousOnload();
    }
    lightboxQuarto.on('slide_before_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      const href = trigger.getAttribute('href');
      if (href !== null) {
        const imgEl = window.document.querySelector(`a[href="${href}"] img`);
        if (imgEl !== null) {
          const srcAttr = imgEl.getAttribute("src");
          if (srcAttr && srcAttr.startsWith("data:")) {
            slideConfig.href = srcAttr;
          }
        }
      } 
    });
  
    lightboxQuarto.on('slide_after_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(slideNode);
      }
    });
  
  };
  
})();
          </script>




</body></html>