<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.40">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>02 - Alinhamento com Cell Ranger ‚Äì Curso de An√°lise Single-Cell</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting-549806ee2085284f45b00abea8c6df48.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap-859d446df91ed362524820d8142de709.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="../../styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../../index.html">
    <span class="navbar-title">Curso de An√°lise Single-Cell</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../index.html"> 
<span class="menu-text">In√≠cio</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../about.html"> 
<span class="menu-text">Sobre</span></a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-semana-1" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Semana 1</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-semana-1">    
        <li>
    <a class="dropdown-item" href="../../material/semana1/01_data_collection.html">
 <span class="dropdown-text">Coleta de Dados</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../material/semana1/02_alignment.html">
 <span class="dropdown-text">Alinhamento</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../material/semana2/03_quality_control.html">
 <span class="dropdown-text">Qualidade de controle</span></a>
  </li>  
    </ul>
  </li>
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#motiva√ß√£o" id="toc-motiva√ß√£o" class="nav-link active" data-scroll-target="#motiva√ß√£o">Motiva√ß√£o</a></li>
  <li><a href="#contextualiza√ß√£o" id="toc-contextualiza√ß√£o" class="nav-link" data-scroll-target="#contextualiza√ß√£o">Contextualiza√ß√£o</a></li>
  <li><a href="#ferramentas-utilizadas" id="toc-ferramentas-utilizadas" class="nav-link" data-scroll-target="#ferramentas-utilizadas">Ferramentas utilizadas</a></li>
  <li><a href="#convers√£o-de-bcl-para-fastq" id="toc-convers√£o-de-bcl-para-fastq" class="nav-link" data-scroll-target="#convers√£o-de-bcl-para-fastq">1. Convers√£o de BCL para FASTQ</a>
  <ul class="collapse">
  <li><a href="#sobre-o-padr√£o-illumina" id="toc-sobre-o-padr√£o-illumina" class="nav-link" data-scroll-target="#sobre-o-padr√£o-illumina">1.1. Sobre o padr√£o Illumina</a></li>
  <li><a href="#sobre-o-padr√£o-10x-genomics" id="toc-sobre-o-padr√£o-10x-genomics" class="nav-link" data-scroll-target="#sobre-o-padr√£o-10x-genomics">1.2. Sobre o padr√£o 10x Genomics</a></li>
  </ul></li>
  <li><a href="#constru√ß√£o-do-genoma-de-refer√™ncia" id="toc-constru√ß√£o-do-genoma-de-refer√™ncia" class="nav-link" data-scroll-target="#constru√ß√£o-do-genoma-de-refer√™ncia">2. Constru√ß√£o do Genoma de Refer√™ncia</a></li>
  <li><a href="#processamento-com-cellranger-count" id="toc-processamento-com-cellranger-count" class="nav-link" data-scroll-target="#processamento-com-cellranger-count">3. Processamento com <code>cellranger count</code></a></li>
  <li><a href="#interpreta√ß√£o-do-relat√≥rio" id="toc-interpreta√ß√£o-do-relat√≥rio" class="nav-link" data-scroll-target="#interpreta√ß√£o-do-relat√≥rio">4. Interpreta√ß√£o do Relat√≥rio</a></li>
  <li><a href="#questionamentos" id="toc-questionamentos" class="nav-link" data-scroll-target="#questionamentos">Questionamentos</a></li>
  <li><a href="#coment√°rios-finais" id="toc-coment√°rios-finais" class="nav-link" data-scroll-target="#coment√°rios-finais">Coment√°rios Finais</a></li>
  <li><a href="#recursos-adicionais" id="toc-recursos-adicionais" class="nav-link" data-scroll-target="#recursos-adicionais">Recursos adicionais</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">02 - Alinhamento com Cell Ranger</h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i> Code</button></div></div>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<section id="motiva√ß√£o" class="level2">
<h2 class="anchored" data-anchor-id="motiva√ß√£o">Motiva√ß√£o</h2>
<p>Ao realizar um experimento de transcript√¥mica de c√©lula √∫nica com a tecnologia 10x Genomics, o sequenciador fornece arquivos no formato BCL. No entanto, o pesquisador precisa de uma matriz de contagem por c√©lula para realizar an√°lises downstream como clustering, anota√ß√£o e detec√ß√£o de tipos celulares.</p>
<p>Este tutorial resolve um problema comum: como transformar dados brutos do sequenciador em uma matriz de express√£o confi√°vel, e como interpretar o relat√≥rio de qualidade do pipeline.</p>
</section>
<section id="contextualiza√ß√£o" class="level2">
<h2 class="anchored" data-anchor-id="contextualiza√ß√£o">Contextualiza√ß√£o</h2>
<p>O <strong>Cell Ranger</strong> √© o pipeline oficial da 10x Genomics para processar dados de scRNA-seq. Ele √© composto por uma s√©rie de ferramentas coordenadas por um framework interno chamado <strong>Martian</strong>, que gerencia etapas paralelas e arquivos intermedi√°rios. O Cell Ranger executa diversas etapas, incluindo demultiplexa√ß√£o, alinhamento, deduplica√ß√£o de UMIs, quantifica√ß√£o g√™nica e gera√ß√£o de relat√≥rios de qualidade. Para instalar o Cell Ranger, por favor, acesso esse <a href="https://www.10xgenomics.com/support/software/cell-ranger/7.2/tutorials/cr-tutorial-in">link</a>.</p>
<blockquote class="blockquote">
<p>üìå Nota: Durante a etapa de alinhamento, o Cell Ranger utiliza o <strong>STAR (Spliced Transcripts Alignment to a Reference)</strong> como seu motor principal para mapear os reads de RNA ao genoma de refer√™ncia. O STAR √© um dos alinhadores mais r√°pidos e sens√≠veis para dados de RNA-seq, sendo particularmente adequado para capturar splicing e isoformas em organismos eucariotos. Caso tenha interesse em entender mais sobre o algoritmo, veja o <a href="https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/1.0/map/cr-counter">link</a></p>
</blockquote>
</section>
<section id="ferramentas-utilizadas" class="level2">
<h2 class="anchored" data-anchor-id="ferramentas-utilizadas">Ferramentas utilizadas</h2>
<p>As etapas do pipeline e suas respectivas fun√ß√µes s√£o:</p>
<ul>
<li><strong><code>cellranger mkfastq</code></strong>: converte arquivos BCL em arquivos FASTQ por amostra.</li>
<li><strong><code>cellranger mkref</code></strong>: constr√≥i um genoma de refer√™ncia com anota√ß√µes para alinhamento.</li>
<li><strong><code>cellranger count</code></strong>: alinha os reads (com STAR), deduplica UMIs, gera matrizes de express√£o e relat√≥rio final de qualidade.</li>
</ul>
</section>
<section id="convers√£o-de-bcl-para-fastq" class="level2">
<h2 class="anchored" data-anchor-id="convers√£o-de-bcl-para-fastq">1. Convers√£o de BCL para FASTQ</h2>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="ex">cellranger</span> mkfastq <span class="dt">\</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">--id</span><span class="op">=</span>Sample001 <span class="dt">\</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">--run</span><span class="op">=</span>/mnt/bcl_run_folder <span class="dt">\</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">--csv</span><span class="op">=</span>sample_sheet.csv</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<ul>
<li>O arquivo <code>sample_sheet.csv</code> deve seguir o padr√£o Illumina, especificando amostras, lanes e barcodes.</li>
</ul>
<section id="sobre-o-padr√£o-illumina" class="level3">
<h3 class="anchored" data-anchor-id="sobre-o-padr√£o-illumina">1.1. Sobre o padr√£o Illumina</h3>
<p>O arquivo CSV segue uma estrutura com cabe√ßalho padronizado e colunas obrigat√≥rias como <code>Sample_ID</code>, <code>Sample_Name</code>, <code>Index</code>, <code>Lane</code>, entre outras. Um exemplo simplificado:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode txt code-with-copy"><code class="sourceCode default"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>[Data]</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>Lane,Sample_ID,Sample_Name,Index</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>1,Sample001,Sample001,SI-GA-A1</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<ul>
<li>A se√ß√£o <code>[Data]</code> delimita o in√≠cio da tabela.</li>
<li><code>Lane</code> N√∫mero da lane do flowcell (geralmente 1, mas pode ser m√∫ltiplas).</li>
<li><code>Sample_ID</code> Identificador interno para a amostra (aparece nos diret√≥rios).</li>
<li><code>Sample_Name</code>Nome descritivo da amostra, que aparece no nome dos arquivos FASTQ.</li>
<li><code>Index</code> Sequ√™ncia ou nome do √≠ndice (barcode) usado para essa amostra.</li>
</ul>
<p>Para mais detalhes, consulte a <a href="https://support.illumina.com/">documenta√ß√£o da Illumina</a>.</p>
</section>
<section id="sobre-o-padr√£o-10x-genomics" class="level3">
<h3 class="anchored" data-anchor-id="sobre-o-padr√£o-10x-genomics">1.2. Sobre o padr√£o 10x Genomics</h3>
<p>Ap√≥s a execu√ß√£o do <code>cellranger mkfastq</code>, os arquivos FASTQ seguem uma conven√ß√£o espec√≠fica da 10x Genomics. Para cada amostra, s√£o gerados arquivos separados por tipo de leitura:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode txt code-with-copy"><code class="sourceCode default"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>bcl2fastq_output</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>|-- HFLC5BBXX</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>    |-- Sample001</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>    |   |-- Sample001_S1_L001_I1_001.fastq.gz</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>    |   |-- Sample001_S1_L001_R1_001.fastq.gz</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>    |   |-- Sample001_S1_L001_R2_001.fastq.gz</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>|-- Reports</span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>|-- Stats</span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>|-- Undetermined_S0_L001_I1_001.fastq.gz</span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>|-- Undetermined_S0_L001_R1_001.fastq.gz</span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a>|-- Undetermined_S0_L001_R2_001.fastq.gz</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Al√©m disso, a nomenclatura do arquivo √© composta por diferentes campos. Veja abaixo:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode txt code-with-copy"><code class="sourceCode default"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>[Sample Name]_S1_L00[Lane Number]_[Read Type]_001.fastq.gz</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Esse nome √© composto por blocos separados por sublinhados (<code>_</code>), com os seguintes significados:</p>
<ul>
<li><strong>Sample001</strong>: nome da amostra, herdado do campo <code>Sample_Name</code> do Sample Sheet.</li>
<li><strong>S1</strong>: identificador da amostra atribu√≠do automaticamente pelo <code>bcl2fastq</code>.</li>
<li><strong>L001</strong>: n√∫mero da lane (neste caso, lane 1).</li>
<li><strong>R1</strong>: tipo da leitura:
<ul>
<li><code>R1</code> = Read 1 (Cell barcode + UMI)</li>
<li><code>R2</code> = Read 2 (cDNA)</li>
<li><code>I1</code> = Index Read 1 (Sample index)</li>
</ul></li>
<li><strong>001</strong>: n√∫mero do set de leitura (geralmente fixo como 001).</li>
</ul>
</section>
</section>
<section id="constru√ß√£o-do-genoma-de-refer√™ncia" class="level2">
<h2 class="anchored" data-anchor-id="constru√ß√£o-do-genoma-de-refer√™ncia">2. Constru√ß√£o do Genoma de Refer√™ncia</h2>
<div class="sourceCode" id="cb5"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="ex">cellranger</span> mkref <span class="dt">\</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">--genome</span><span class="op">=</span>GRCh38_custom <span class="dt">\</span></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">--fasta</span><span class="op">=</span>genome.fa <span class="dt">\</span></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">--genes</span><span class="op">=</span>genes.gtf</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<ul>
<li>Entrada: arquivos <code>FASTA</code> com a sequ√™ncia do genoma e <code>GTF</code> com as anota√ß√µes de genes.</li>
<li>Sa√≠da: diret√≥rio contendo a refer√™ncia indexada para uso posterior.</li>
</ul>
</section>
<section id="processamento-com-cellranger-count" class="level2">
<h2 class="anchored" data-anchor-id="processamento-com-cellranger-count">3. Processamento com <code>cellranger count</code></h2>
<div class="sourceCode" id="cb6"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="ex">cellranger</span> count <span class="dt">\</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">--id</span><span class="op">=</span>Sample001 <span class="dt">\</span></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">--transcriptome</span><span class="op">=</span>/refs/GRCh38_custom <span class="dt">\</span></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">--fastqs</span><span class="op">=</span>/mnt/fastqs <span class="dt">\</span></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">--sample</span><span class="op">=</span>Sample001</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Esta etapa realiza:</p>
<ul>
<li>Alinhamento com STAR.</li>
<li>Identifica√ß√£o de c√©lulas por barcodes √∫nicos.</li>
<li>Quantifica√ß√£o g√™nica por UMI.</li>
<li>Gera√ß√£o das matrizes de contagem e do relat√≥rio de qualidade (<code>web_summary.html</code>).</li>
</ul>
</section>
<section id="interpreta√ß√£o-do-relat√≥rio" class="level2">
<h2 class="anchored" data-anchor-id="interpreta√ß√£o-do-relat√≥rio">4. Interpreta√ß√£o do Relat√≥rio</h2>
<p>O Cell Ranger gera um relat√≥rio interativo (<code>web_summary.html</code>) com gr√°ficos e tabelas que ajudam a avaliar a qualidade do sequenciamento e da captura celular.</p>
<p>Principais m√©tricas e suas interpreta√ß√µes:</p>
<ul>
<li><strong>Estimated Number of Cells</strong>: n√∫mero estimado de c√©lulas √∫nicas com base na distribui√ß√£o de UMIs.</li>
<li><strong>Reads Mapped to Genome</strong>: porcentagem de reads que alinharam ao genoma de refer√™ncia (ideal &gt; 80%).</li>
<li><strong>Sequencing Saturation</strong>: medida de redund√¢ncia dos reads (ideal &gt; 40%); valores baixos sugerem que mais sequenciamento pode ser necess√°rio.</li>
<li><strong>Fraction Reads in Cells</strong>: fra√ß√£o de reads atribu√≠das a barcodes v√°lidos (ideal &gt; 70%).</li>
<li><strong>Valid Barcodes</strong>: porcentagem de barcodes reconhecidos como v√°lidos (ideal &gt; 90%).</li>
<li><strong>Median Genes per Cell</strong>: n√∫mero mediano de genes detectados por c√©lula; depende do tipo celular e qualidade da amostra.</li>
</ul>
</section>
<section id="questionamentos" class="level2">
<h2 class="anchored" data-anchor-id="questionamentos">Questionamentos</h2>
<ul>
<li>O n√∫mero de c√©lulas detectadas est√° de acordo com a expectativa para o protocolo de captura utilizado?</li>
<li>H√° evid√™ncia de RNA ambiente ou c√©lulas estressadas (baixa fra√ß√£o de reads em c√©lulas)?</li>
<li>A satura√ß√£o sugere que o sequenciamento foi suficiente ou deveria ser mais profundo?</li>
<li>A distribui√ß√£o dos genes por c√©lula est√° coerente com o tipo de tecido ou condi√ß√£o estudada?</li>
<li>Os barcodes inv√°lidos est√£o ocorrendo em excesso, indicando problemas na biblioteca?</li>
</ul>
</section>
<section id="coment√°rios-finais" class="level2">
<h2 class="anchored" data-anchor-id="coment√°rios-finais">Coment√°rios Finais</h2>
<p>Este tutorial demonstrou o uso do Cell Ranger para:</p>
<ul>
<li>Processar dados brutos (BCL) at√© a gera√ß√£o das matrizes de contagem por c√©lula.</li>
<li>Avaliar a qualidade dos dados utilizando o relat√≥rio interativo fornecido pelo pr√≥prio pipeline.</li>
<li>Refletir criticamente sobre os resultados, visando decis√µes informadas para an√°lises downstream.</li>
</ul>
<p>A compreens√£o dessa etapa √© essencial para garantir que an√°lises como clustering, infer√™ncia de trajet√≥rias e anota√ß√£o celular sejam conduzidas com dados confi√°veis e bem interpretados.</p>
</section>
<section id="recursos-adicionais" class="level2">
<h2 class="anchored" data-anchor-id="recursos-adicionais">Recursos adicionais</h2>
<ul>
<li><a href="https://support.10xgenomics.com/single-cell-gene-expression">Documenta√ß√£o oficial</a></li>
<li><a href="https://satijalab.org/seurat/articles/pbmc3k_tutorial.html">Tutorial de QC no Seurat</a></li>
<li><a href="https://github.com/alexdobin/STAR">STAR aligner</a></li>
</ul>


<!-- -->

</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "Óßã";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp("https:\/\/lint-edu\.github\.io\/PPGBM0117\.2025\.1");
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb7" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "02 - Alinhamento com Cell Ranger"</span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="an">format:</span></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a><span class="co">  html:</span></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a><span class="co">    toc: true</span></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a><span class="co">    toc-depth: 3</span></span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a><span class="co">    code-fold: true</span></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a><span class="co">    code-tools: true</span></span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a><span class="fu">## Motiva√ß√£o</span></span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a>Ao realizar um experimento de transcript√¥mica de c√©lula √∫nica com a tecnologia 10x Genomics, o sequenciador fornece arquivos no formato BCL. No entanto, o pesquisador precisa de uma matriz de contagem por c√©lula para realizar an√°lises downstream como clustering, anota√ß√£o e detec√ß√£o de tipos celulares.</span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a>Este tutorial resolve um problema comum: como transformar dados brutos do sequenciador em uma matriz de express√£o confi√°vel, e como interpretar o relat√≥rio de qualidade do pipeline.</span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-17"><a href="#cb7-17" aria-hidden="true" tabindex="-1"></a><span class="fu">## Contextualiza√ß√£o</span></span>
<span id="cb7-18"><a href="#cb7-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-19"><a href="#cb7-19" aria-hidden="true" tabindex="-1"></a>O **Cell Ranger** √© o pipeline oficial da 10x Genomics para processar dados de scRNA-seq. Ele √© composto por uma s√©rie de ferramentas coordenadas por um framework interno chamado **Martian**, que gerencia etapas paralelas e arquivos intermedi√°rios. O Cell Ranger executa diversas etapas, incluindo demultiplexa√ß√£o, alinhamento, deduplica√ß√£o de UMIs, quantifica√ß√£o g√™nica e gera√ß√£o de relat√≥rios de qualidade. Para instalar o Cell Ranger, por favor, acesso esse <span class="co">[</span><span class="ot">link</span><span class="co">](https://www.10xgenomics.com/support/software/cell-ranger/7.2/tutorials/cr-tutorial-in)</span>. </span>
<span id="cb7-20"><a href="#cb7-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-21"><a href="#cb7-21" aria-hidden="true" tabindex="-1"></a><span class="at">&gt; üìå Nota: Durante a etapa de alinhamento, o Cell Ranger utiliza o **STAR (Spliced Transcripts Alignment to a Reference)** como seu motor principal para mapear os reads de RNA ao genoma de refer√™ncia. O STAR √© um dos alinhadores mais r√°pidos e sens√≠veis para dados de RNA-seq, sendo particularmente adequado para capturar splicing e isoformas em organismos eucariotos. Caso tenha interesse em entender mais sobre o algoritmo, veja o </span><span class="co">[</span><span class="ot">link</span><span class="co">](https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/1.0/map/cr-counter)</span></span>
<span id="cb7-22"><a href="#cb7-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-23"><a href="#cb7-23" aria-hidden="true" tabindex="-1"></a><span class="fu">## Ferramentas utilizadas</span></span>
<span id="cb7-24"><a href="#cb7-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-25"><a href="#cb7-25" aria-hidden="true" tabindex="-1"></a>As etapas do pipeline e suas respectivas fun√ß√µes s√£o:</span>
<span id="cb7-26"><a href="#cb7-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-27"><a href="#cb7-27" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>**`cellranger mkfastq`**: converte arquivos BCL em arquivos FASTQ por amostra.</span>
<span id="cb7-28"><a href="#cb7-28" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>**`cellranger mkref`**: constr√≥i um genoma de refer√™ncia com anota√ß√µes para alinhamento.</span>
<span id="cb7-29"><a href="#cb7-29" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>**`cellranger count`**: alinha os reads (com STAR), deduplica UMIs, gera matrizes de express√£o e relat√≥rio final de qualidade.</span>
<span id="cb7-30"><a href="#cb7-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-31"><a href="#cb7-31" aria-hidden="true" tabindex="-1"></a><span class="fu">## 1. Convers√£o de BCL para FASTQ</span></span>
<span id="cb7-32"><a href="#cb7-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-33"><a href="#cb7-33" aria-hidden="true" tabindex="-1"></a><span class="in">```bash</span></span>
<span id="cb7-34"><a href="#cb7-34" aria-hidden="true" tabindex="-1"></a><span class="ex">cellranger</span> mkfastq <span class="dt">\</span></span>
<span id="cb7-35"><a href="#cb7-35" aria-hidden="true" tabindex="-1"></a>  <span class="at">--id</span><span class="op">=</span>Sample001 <span class="dt">\</span></span>
<span id="cb7-36"><a href="#cb7-36" aria-hidden="true" tabindex="-1"></a>  <span class="at">--run</span><span class="op">=</span>/mnt/bcl_run_folder <span class="dt">\</span></span>
<span id="cb7-37"><a href="#cb7-37" aria-hidden="true" tabindex="-1"></a>  <span class="at">--csv</span><span class="op">=</span>sample_sheet.csv</span>
<span id="cb7-38"><a href="#cb7-38" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-39"><a href="#cb7-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-40"><a href="#cb7-40" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>O arquivo <span class="in">`sample_sheet.csv`</span> deve seguir o padr√£o Illumina, especificando amostras, lanes e barcodes.</span>
<span id="cb7-41"><a href="#cb7-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-42"><a href="#cb7-42" aria-hidden="true" tabindex="-1"></a><span class="fu">### 1.1. Sobre o padr√£o Illumina</span></span>
<span id="cb7-43"><a href="#cb7-43" aria-hidden="true" tabindex="-1"></a>O arquivo CSV segue uma estrutura com cabe√ßalho padronizado e colunas obrigat√≥rias como <span class="in">`Sample_ID`</span>, <span class="in">`Sample_Name`</span>, <span class="in">`Index`</span>, <span class="in">`Lane`</span>, entre outras. Um exemplo simplificado:</span>
<span id="cb7-44"><a href="#cb7-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-45"><a href="#cb7-45" aria-hidden="true" tabindex="-1"></a><span class="in">```txt</span></span>
<span id="cb7-46"><a href="#cb7-46" aria-hidden="true" tabindex="-1"></a><span class="in">[Data]</span></span>
<span id="cb7-47"><a href="#cb7-47" aria-hidden="true" tabindex="-1"></a><span class="in">Lane,Sample_ID,Sample_Name,Index</span></span>
<span id="cb7-48"><a href="#cb7-48" aria-hidden="true" tabindex="-1"></a><span class="in">1,Sample001,Sample001,SI-GA-A1</span></span>
<span id="cb7-49"><a href="#cb7-49" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-50"><a href="#cb7-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-51"><a href="#cb7-51" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>A se√ß√£o <span class="in">`[Data]`</span> delimita o in√≠cio da tabela.</span>
<span id="cb7-52"><a href="#cb7-52" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span><span class="in">`Lane`</span> N√∫mero da lane do flowcell (geralmente 1, mas pode ser m√∫ltiplas).</span>
<span id="cb7-53"><a href="#cb7-53" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span><span class="in">`Sample_ID`</span> Identificador interno para a amostra (aparece nos diret√≥rios).</span>
<span id="cb7-54"><a href="#cb7-54" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span><span class="in">`Sample_Name`</span>Nome descritivo da amostra, que aparece no nome dos arquivos FASTQ.</span>
<span id="cb7-55"><a href="#cb7-55" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span><span class="in">`Index`</span> Sequ√™ncia ou nome do √≠ndice (barcode) usado para essa amostra.</span>
<span id="cb7-56"><a href="#cb7-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-57"><a href="#cb7-57" aria-hidden="true" tabindex="-1"></a>Para mais detalhes, consulte a <span class="co">[</span><span class="ot">documenta√ß√£o da Illumina</span><span class="co">](https://support.illumina.com/)</span>. </span>
<span id="cb7-58"><a href="#cb7-58" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-59"><a href="#cb7-59" aria-hidden="true" tabindex="-1"></a><span class="fu">### 1.2. Sobre o padr√£o 10x Genomics</span></span>
<span id="cb7-60"><a href="#cb7-60" aria-hidden="true" tabindex="-1"></a>Ap√≥s a execu√ß√£o do <span class="in">`cellranger mkfastq`</span>, os arquivos FASTQ seguem uma conven√ß√£o espec√≠fica da 10x Genomics. Para cada amostra, s√£o gerados arquivos separados por tipo de leitura:</span>
<span id="cb7-61"><a href="#cb7-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-62"><a href="#cb7-62" aria-hidden="true" tabindex="-1"></a><span class="in">```txt</span></span>
<span id="cb7-63"><a href="#cb7-63" aria-hidden="true" tabindex="-1"></a><span class="in">bcl2fastq_output</span></span>
<span id="cb7-64"><a href="#cb7-64" aria-hidden="true" tabindex="-1"></a><span class="in">|-- HFLC5BBXX</span></span>
<span id="cb7-65"><a href="#cb7-65" aria-hidden="true" tabindex="-1"></a><span class="in">    |-- Sample001</span></span>
<span id="cb7-66"><a href="#cb7-66" aria-hidden="true" tabindex="-1"></a><span class="in">    |   |-- Sample001_S1_L001_I1_001.fastq.gz</span></span>
<span id="cb7-67"><a href="#cb7-67" aria-hidden="true" tabindex="-1"></a><span class="in">    |   |-- Sample001_S1_L001_R1_001.fastq.gz</span></span>
<span id="cb7-68"><a href="#cb7-68" aria-hidden="true" tabindex="-1"></a><span class="in">    |   |-- Sample001_S1_L001_R2_001.fastq.gz</span></span>
<span id="cb7-69"><a href="#cb7-69" aria-hidden="true" tabindex="-1"></a><span class="in">|-- Reports</span></span>
<span id="cb7-70"><a href="#cb7-70" aria-hidden="true" tabindex="-1"></a><span class="in">|-- Stats</span></span>
<span id="cb7-71"><a href="#cb7-71" aria-hidden="true" tabindex="-1"></a><span class="in">|-- Undetermined_S0_L001_I1_001.fastq.gz</span></span>
<span id="cb7-72"><a href="#cb7-72" aria-hidden="true" tabindex="-1"></a><span class="in">|-- Undetermined_S0_L001_R1_001.fastq.gz</span></span>
<span id="cb7-73"><a href="#cb7-73" aria-hidden="true" tabindex="-1"></a><span class="in">|-- Undetermined_S0_L001_R2_001.fastq.gz</span></span>
<span id="cb7-74"><a href="#cb7-74" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-75"><a href="#cb7-75" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-76"><a href="#cb7-76" aria-hidden="true" tabindex="-1"></a>Al√©m disso, a nomenclatura do arquivo √© composta por diferentes campos. Veja abaixo:</span>
<span id="cb7-77"><a href="#cb7-77" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-78"><a href="#cb7-78" aria-hidden="true" tabindex="-1"></a><span class="in">```txt</span></span>
<span id="cb7-79"><a href="#cb7-79" aria-hidden="true" tabindex="-1"></a><span class="in">[Sample Name]_S1_L00[Lane Number]_[Read Type]_001.fastq.gz</span></span>
<span id="cb7-80"><a href="#cb7-80" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-81"><a href="#cb7-81" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-82"><a href="#cb7-82" aria-hidden="true" tabindex="-1"></a>Esse nome √© composto por blocos separados por sublinhados (<span class="in">`_`</span>), com os seguintes significados:</span>
<span id="cb7-83"><a href="#cb7-83" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-84"><a href="#cb7-84" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>**Sample001**: nome da amostra, herdado do campo <span class="in">`Sample_Name`</span> do Sample Sheet.</span>
<span id="cb7-85"><a href="#cb7-85" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>**S1**: identificador da amostra atribu√≠do automaticamente pelo <span class="in">`bcl2fastq`</span>.</span>
<span id="cb7-86"><a href="#cb7-86" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>**L001**: n√∫mero da lane (neste caso, lane 1).</span>
<span id="cb7-87"><a href="#cb7-87" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>**R1**: tipo da leitura:</span>
<span id="cb7-88"><a href="#cb7-88" aria-hidden="true" tabindex="-1"></a><span class="ss">  - </span><span class="in">`R1`</span> = Read 1 (Cell barcode + UMI)</span>
<span id="cb7-89"><a href="#cb7-89" aria-hidden="true" tabindex="-1"></a><span class="ss">  - </span><span class="in">`R2`</span> = Read 2 (cDNA)</span>
<span id="cb7-90"><a href="#cb7-90" aria-hidden="true" tabindex="-1"></a><span class="ss">  - </span><span class="in">`I1`</span> = Index Read 1 (Sample index)</span>
<span id="cb7-91"><a href="#cb7-91" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>**001**: n√∫mero do set de leitura (geralmente fixo como 001).</span>
<span id="cb7-92"><a href="#cb7-92" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-93"><a href="#cb7-93" aria-hidden="true" tabindex="-1"></a><span class="fu">## 2. Constru√ß√£o do Genoma de Refer√™ncia</span></span>
<span id="cb7-94"><a href="#cb7-94" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-95"><a href="#cb7-95" aria-hidden="true" tabindex="-1"></a><span class="in">```bash</span></span>
<span id="cb7-96"><a href="#cb7-96" aria-hidden="true" tabindex="-1"></a><span class="ex">cellranger</span> mkref <span class="dt">\</span></span>
<span id="cb7-97"><a href="#cb7-97" aria-hidden="true" tabindex="-1"></a>  <span class="at">--genome</span><span class="op">=</span>GRCh38_custom <span class="dt">\</span></span>
<span id="cb7-98"><a href="#cb7-98" aria-hidden="true" tabindex="-1"></a>  <span class="at">--fasta</span><span class="op">=</span>genome.fa <span class="dt">\</span></span>
<span id="cb7-99"><a href="#cb7-99" aria-hidden="true" tabindex="-1"></a>  <span class="at">--genes</span><span class="op">=</span>genes.gtf</span>
<span id="cb7-100"><a href="#cb7-100" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-101"><a href="#cb7-101" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-102"><a href="#cb7-102" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Entrada: arquivos <span class="in">`FASTA`</span> com a sequ√™ncia do genoma e <span class="in">`GTF`</span> com as anota√ß√µes de genes.</span>
<span id="cb7-103"><a href="#cb7-103" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Sa√≠da: diret√≥rio contendo a refer√™ncia indexada para uso posterior.</span>
<span id="cb7-104"><a href="#cb7-104" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-105"><a href="#cb7-105" aria-hidden="true" tabindex="-1"></a><span class="fu">## 3. Processamento com `cellranger count`</span></span>
<span id="cb7-106"><a href="#cb7-106" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-107"><a href="#cb7-107" aria-hidden="true" tabindex="-1"></a><span class="in">```bash</span></span>
<span id="cb7-108"><a href="#cb7-108" aria-hidden="true" tabindex="-1"></a><span class="ex">cellranger</span> count <span class="dt">\</span></span>
<span id="cb7-109"><a href="#cb7-109" aria-hidden="true" tabindex="-1"></a>  <span class="at">--id</span><span class="op">=</span>Sample001 <span class="dt">\</span></span>
<span id="cb7-110"><a href="#cb7-110" aria-hidden="true" tabindex="-1"></a>  <span class="at">--transcriptome</span><span class="op">=</span>/refs/GRCh38_custom <span class="dt">\</span></span>
<span id="cb7-111"><a href="#cb7-111" aria-hidden="true" tabindex="-1"></a>  <span class="at">--fastqs</span><span class="op">=</span>/mnt/fastqs <span class="dt">\</span></span>
<span id="cb7-112"><a href="#cb7-112" aria-hidden="true" tabindex="-1"></a>  <span class="at">--sample</span><span class="op">=</span>Sample001</span>
<span id="cb7-113"><a href="#cb7-113" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb7-114"><a href="#cb7-114" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-115"><a href="#cb7-115" aria-hidden="true" tabindex="-1"></a>Esta etapa realiza:</span>
<span id="cb7-116"><a href="#cb7-116" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-117"><a href="#cb7-117" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Alinhamento com STAR.</span>
<span id="cb7-118"><a href="#cb7-118" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Identifica√ß√£o de c√©lulas por barcodes √∫nicos.</span>
<span id="cb7-119"><a href="#cb7-119" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Quantifica√ß√£o g√™nica por UMI.</span>
<span id="cb7-120"><a href="#cb7-120" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Gera√ß√£o das matrizes de contagem e do relat√≥rio de qualidade (<span class="in">`web_summary.html`</span>).</span>
<span id="cb7-121"><a href="#cb7-121" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-122"><a href="#cb7-122" aria-hidden="true" tabindex="-1"></a><span class="fu">## 4. Interpreta√ß√£o do Relat√≥rio</span></span>
<span id="cb7-123"><a href="#cb7-123" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-124"><a href="#cb7-124" aria-hidden="true" tabindex="-1"></a>O Cell Ranger gera um relat√≥rio interativo (<span class="in">`web_summary.html`</span>) com gr√°ficos e tabelas que ajudam a avaliar a qualidade do sequenciamento e da captura celular.</span>
<span id="cb7-125"><a href="#cb7-125" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-126"><a href="#cb7-126" aria-hidden="true" tabindex="-1"></a>Principais m√©tricas e suas interpreta√ß√µes:</span>
<span id="cb7-127"><a href="#cb7-127" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-128"><a href="#cb7-128" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>**Estimated Number of Cells**: n√∫mero estimado de c√©lulas √∫nicas com base na distribui√ß√£o de UMIs.</span>
<span id="cb7-129"><a href="#cb7-129" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>**Reads Mapped to Genome**: porcentagem de reads que alinharam ao genoma de refer√™ncia (ideal &gt; 80%).</span>
<span id="cb7-130"><a href="#cb7-130" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>**Sequencing Saturation**: medida de redund√¢ncia dos reads (ideal &gt; 40%); valores baixos sugerem que mais sequenciamento pode ser necess√°rio.</span>
<span id="cb7-131"><a href="#cb7-131" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>**Fraction Reads in Cells**: fra√ß√£o de reads atribu√≠das a barcodes v√°lidos (ideal &gt; 70%).</span>
<span id="cb7-132"><a href="#cb7-132" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>**Valid Barcodes**: porcentagem de barcodes reconhecidos como v√°lidos (ideal &gt; 90%).</span>
<span id="cb7-133"><a href="#cb7-133" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>**Median Genes per Cell**: n√∫mero mediano de genes detectados por c√©lula; depende do tipo celular e qualidade da amostra.</span>
<span id="cb7-134"><a href="#cb7-134" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-135"><a href="#cb7-135" aria-hidden="true" tabindex="-1"></a><span class="fu">## Questionamentos</span></span>
<span id="cb7-136"><a href="#cb7-136" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-137"><a href="#cb7-137" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>O n√∫mero de c√©lulas detectadas est√° de acordo com a expectativa para o protocolo de captura utilizado?</span>
<span id="cb7-138"><a href="#cb7-138" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>H√° evid√™ncia de RNA ambiente ou c√©lulas estressadas (baixa fra√ß√£o de reads em c√©lulas)?</span>
<span id="cb7-139"><a href="#cb7-139" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>A satura√ß√£o sugere que o sequenciamento foi suficiente ou deveria ser mais profundo?</span>
<span id="cb7-140"><a href="#cb7-140" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>A distribui√ß√£o dos genes por c√©lula est√° coerente com o tipo de tecido ou condi√ß√£o estudada?</span>
<span id="cb7-141"><a href="#cb7-141" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Os barcodes inv√°lidos est√£o ocorrendo em excesso, indicando problemas na biblioteca?</span>
<span id="cb7-142"><a href="#cb7-142" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-143"><a href="#cb7-143" aria-hidden="true" tabindex="-1"></a><span class="fu">## Coment√°rios Finais</span></span>
<span id="cb7-144"><a href="#cb7-144" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-145"><a href="#cb7-145" aria-hidden="true" tabindex="-1"></a>Este tutorial demonstrou o uso do Cell Ranger para:</span>
<span id="cb7-146"><a href="#cb7-146" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-147"><a href="#cb7-147" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Processar dados brutos (BCL) at√© a gera√ß√£o das matrizes de contagem por c√©lula.</span>
<span id="cb7-148"><a href="#cb7-148" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Avaliar a qualidade dos dados utilizando o relat√≥rio interativo fornecido pelo pr√≥prio pipeline.</span>
<span id="cb7-149"><a href="#cb7-149" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Refletir criticamente sobre os resultados, visando decis√µes informadas para an√°lises downstream.</span>
<span id="cb7-150"><a href="#cb7-150" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-151"><a href="#cb7-151" aria-hidden="true" tabindex="-1"></a>A compreens√£o dessa etapa √© essencial para garantir que an√°lises como clustering, infer√™ncia de trajet√≥rias e anota√ß√£o celular sejam conduzidas com dados confi√°veis e bem interpretados.</span>
<span id="cb7-152"><a href="#cb7-152" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-153"><a href="#cb7-153" aria-hidden="true" tabindex="-1"></a><span class="fu">## Recursos adicionais</span></span>
<span id="cb7-154"><a href="#cb7-154" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-155"><a href="#cb7-155" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span><span class="co">[</span><span class="ot">Documenta√ß√£o oficial</span><span class="co">](https://support.10xgenomics.com/single-cell-gene-expression)</span></span>
<span id="cb7-156"><a href="#cb7-156" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span><span class="co">[</span><span class="ot">Tutorial de QC no Seurat</span><span class="co">](https://satijalab.org/seurat/articles/pbmc3k_tutorial.html)</span></span>
<span id="cb7-157"><a href="#cb7-157" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span><span class="co">[</span><span class="ot">STAR aligner</span><span class="co">](https://github.com/alexdobin/STAR)</span> </span></code><button title="Copy to Clipboard" class="code-copy-button" data-in-quarto-modal=""><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">
      &nbsp;
    </div>   
    <div class="nav-footer-center">
<p>Feito com ‚ù§Ô∏è usando <a href="https://quarto.org">Quarto</a></p>
</div>
    <div class="nav-footer-right">
      &nbsp;
    </div>
  </div>
</footer>




</body></html>